From 18a5bccf3ea510c548f10c4ec1c1c2904dce0250 Mon Sep 17 00:00:00 2001
From: Mian Yousaf Kaukab <yousaf.kaukab@suse.com>
Date: Fri, 6 Oct 2017 11:30:34 +0200
Subject: [PATCH 00/56] perf: arm: temporary workaround for build errors

Patch-mainline: Never, Workaround to fix build errors
References: bsc#1062279

Signed-off-by: Mian Yousaf Kaukab <yousaf.kaukab@suse.com>
---
 include/linux/perf/arm_pmu.h | 60 ++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 60 insertions(+)

diff --git a/include/linux/perf/arm_pmu.h b/include/linux/perf/arm_pmu.h
index fc3d3981..2fec5978 100644
--- a/include/linux/perf/arm_pmu.h
+++ b/include/linux/perf/arm_pmu.h
@@ -18,6 +18,66 @@
 #include <asm/cputype.h>
 
 /*
+ * Temporary build fixes
+ */
+static inline int irq_get_percpu_devid_partition(unsigned int irq,
+		struct cpumask *affinity)
+{
+	return -ENOTSUPP;
+}
+
+static inline int cpuhp_setup_state_multi(enum cpuhp_state state,
+					  const char *name,
+					  int (*startup)(unsigned int cpu,
+							 struct hlist_node *node),
+					  int (*teardown)(unsigned int cpu,
+							  struct hlist_node *node))
+{
+	return -ENOTSUPP;
+}
+
+static inline int cpuhp_setup_state_nocalls(enum cpuhp_state state,
+					    const char *name,
+					    int (*startup)(unsigned int cpu),
+					    int (*teardown)(unsigned int cpu))
+{
+	return -ENOTSUPP;
+}
+
+static inline int cpuhp_setup_state(enum cpuhp_state state,
+				    const char *name,
+				    int (*startup)(unsigned int cpu),
+				    int (*teardown)(unsigned int cpu))
+{
+	return -ENOTSUPP;
+}
+
+static inline void cpuhp_remove_state(enum cpuhp_state state) { }
+
+static inline void cpuhp_remove_state_nocalls(enum cpuhp_state state) { }
+
+static inline int cpuhp_state_remove_instance_nocalls(enum cpuhp_state state,
+						      struct hlist_node *node)
+{
+	return -ENOTSUPP;
+}
+
+static inline int cpuhp_state_add_instance(enum cpuhp_state state,
+					   struct hlist_node *node)
+{
+	return -ENOTSUPP;
+}
+
+static inline int cpuhp_state_add_instance_nocalls(enum cpuhp_state state,
+						   struct hlist_node *node)
+{
+	return -ENOTSUPP;
+}
+
+#define CPUHP_AP_PERF_ARM_STARTING 0
+#define CPUHP_AP_PERF_ARM_ACPI_STARTING 1
+
+/*
  * struct arm_pmu_platdata - ARM PMU platform data
  *
  * @handle_irq: an optional handler which will be called from the
-- 
2.11.0

