From: Jitendra Kalsaria <jitendra.kalsaria@qlogic.com>
Date: Wed, 18 Aug 2010 18:40:26 -0700
Subject: [PATCH 02/14] qlge: Changing cpu_to_be16 to htons for udp checksum.
Patch-mainline: No
Git-commit: 9478d2793931bbd77b0e86c0cecf2b475d283ac7

Signed-off-by: Jitendra Kalsaria <jitendra.kalsaria@qlogic.com>
Acked-by: Jeff Mahoney <jeffm@suse.com>
---
 drivers/net/qlge/qlge_main.c |   17 +++++++----------
 1 files changed, 7 insertions(+), 10 deletions(-)

diff --git a/drivers/net/qlge/qlge_main.c b/drivers/net/qlge/qlge_main.c
index fe98373..6e3b903 100644
--- a/drivers/net/qlge/qlge_main.c
+++ b/drivers/net/qlge/qlge_main.c
@@ -1557,11 +1557,10 @@ static void ql_process_mac_rx_page(struct ql_adapter *qdev,
 				(ib_mac_rsp->flags3 & IB_MAC_IOCB_RSP_V4)) {
 			/* Unfragmented ipv4 UDP frame. */
 			struct iphdr *iph = (struct iphdr *) skb->data;
-			if (!(iph->frag_off &
-				cpu_to_be16(IP_MF|IP_OFFSET))) {
+			if (!(iph->frag_off & htons(IP_MF|IP_OFFSET))) {
 				skb->ip_summed = CHECKSUM_UNNECESSARY;
 				QPRINTK(qdev, RX_STATUS, DEBUG,
-						"TCP checksum done!\n");
+						"UDP checksum done!\n");
 			}
 		}
 	}
@@ -1667,11 +1666,10 @@ static void ql_process_mac_rx_skb(struct ql_adapter *qdev,
 				(ib_mac_rsp->flags3 & IB_MAC_IOCB_RSP_V4)) {
 			/* Unfragmented ipv4 UDP frame. */
 			struct iphdr *iph = (struct iphdr *) skb->data;
-			if (!(iph->frag_off &
-				cpu_to_be16(IP_MF|IP_OFFSET))) {
+			if (!(iph->frag_off & htons(IP_MF|IP_OFFSET))) {
 				skb->ip_summed = CHECKSUM_UNNECESSARY;
 				QPRINTK(qdev, RX_STATUS, DEBUG,
-						"TCP checksum done!\n");
+						"UDP checksum done!\n");
 			}
 		}
 	}
@@ -1978,13 +1976,12 @@ static void ql_process_mac_split_rx_intr(struct ql_adapter *qdev,
 			skb->ip_summed = CHECKSUM_UNNECESSARY;
 		} else if ((ib_mac_rsp->flags2 & IB_MAC_IOCB_RSP_U) &&
 				(ib_mac_rsp->flags3 & IB_MAC_IOCB_RSP_V4)) {
-		/* Unfragmented ipv4 UDP frame. */
+			/* Unfragmented ipv4 UDP frame. */
 			struct iphdr *iph = (struct iphdr *) skb->data;
-			if (!(iph->frag_off &
-				cpu_to_be16(IP_MF|IP_OFFSET))) {
+			if (!(iph->frag_off & htons(IP_MF|IP_OFFSET))) {
 				skb->ip_summed = CHECKSUM_UNNECESSARY;
 				QPRINTK(qdev, RX_STATUS, DEBUG,
-						"TCP checksum done!\n");
+						"UDP checksum done!\n");
 			}
 		}
 	}
-- 
1.7.6.rc0.12.g2c6b5

