From db71336b9eec22c21cef65c90cea49130c464994 Mon Sep 17 00:00:00 2001
From: Philipp Zabel <p.zabel@pengutronix.de>
Date: Fri, 22 Apr 2016 10:40:11 +0200
Subject: [PATCH] ASoC: hdmi-codec: Add ELD control
Git-commit: db71336b9eec22c21cef65c90cea49130c464994
Patch-mainline: v4.7-rc1
References: fate#322003

Signed-off-by: Mark Brown <broonie@kernel.org>

Signed-off-by: Patrik Jakobsson <pjakobsson@suse.de>
---
 sound/soc/codecs/hdmi-codec.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/sound/soc/codecs/hdmi-codec.c b/sound/soc/codecs/hdmi-codec.c
index c78333b4311d..8e36e883e453 100644
--- a/sound/soc/codecs/hdmi-codec.c
+++ b/sound/soc/codecs/hdmi-codec.c
@@ -65,9 +65,7 @@ static int hdmi_eld_ctl_get(struct snd_kcontrol *kcontrol,
 	struct snd_soc_component *component = snd_kcontrol_chip(kcontrol);
 	struct hdmi_codec_priv *hcp = snd_soc_component_get_drvdata(component);
 
-	mutex_lock(&hcp->eld_lock);
 	memcpy(ucontrol->value.bytes.data, hcp->eld, sizeof(hcp->eld));
-	mutex_unlock(&hcp->eld_lock);
 
 	return 0;
 }
-- 
2.12.2
