From: Jitendra Kalsaria <jitendra.kalsaria@qlogic.com>
Date: Fri, 15 Oct 2010 23:09:13 -0700
Subject: [PATCH 11/14] qlge: pull NULL check ahead of dereference.
Git-commit: 4da79504b2b41410dd56e9268c9ad403aa65e89c
Patch-mainline: v2.6.37-rc1

Signed-off-by: Jitendra Kalsaria <jitendra.kalsaria@qlogic.com>
Acked-by: Jeff Mahoney <jeffm@suse.com>
---
 drivers/net/qlge/qlge_main.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/net/qlge/qlge_main.c b/drivers/net/qlge/qlge_main.c
index 16624b6..035aea2 100644
--- a/drivers/net/qlge/qlge_main.c
+++ b/drivers/net/qlge/qlge_main.c
@@ -2197,10 +2197,11 @@ static int ql_clean_outbound_rx_ring(struct rx_ring *rx_ring)
 		ql_update_cq(rx_ring);
 		prod = ql_read_sh_reg(rx_ring->prod_idx_sh_reg);
 	}
+	if (!net_rsp)
+		return 0;
 	ql_write_cq_idx(rx_ring);
 	tx_ring = &qdev->tx_ring[net_rsp->txq_idx];
-	if (__netif_subqueue_stopped(qdev->ndev, tx_ring->wq_id) &&
-					net_rsp != NULL) {
+	if (__netif_subqueue_stopped(qdev->ndev, tx_ring->wq_id)) {
 		if (atomic_read(&tx_ring->queue_stopped) &&
 		    (atomic_read(&tx_ring->tx_count) > (tx_ring->wq_len / 4)))
 			/*
-- 
1.7.6.rc0.12.g2c6b5

