From: Vinod Koul <vinod.koul@intel.com>
Date: Tue, 12 Jul 2016 10:09:15 +0530
Subject: dmaengine: mv_xor_v2: remove trailing whitespace
Git-commit: f1532019dd5b2a9c5b659896968626e6e8fb232e
Patch-mainline: v4.8-rc1
References: fate#319899 fate#320700

Signed-off-by: Vinod Koul <vinod.koul@intel.com>
Signed-off-by: Matthias Brugger <mbrugger@suse.com>
---
 drivers/dma/mv_xor_v2.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/dma/mv_xor_v2.c b/drivers/dma/mv_xor_v2.c
index 8320155..a28a01f 100644
--- a/drivers/dma/mv_xor_v2.c
+++ b/drivers/dma/mv_xor_v2.c
@@ -876,4 +876,3 @@ module_platform_driver(mv_xor_v2_driver);
 
 MODULE_DESCRIPTION("DMA engine driver for Marvell's Version 2 of XOR engine");
 MODULE_LICENSE("GPL");
-
-- 
2.10.0

