From sarveshwarb@serverengines.com  Thu Jun 23 15:40:24 2011
From: Sarveshwar Bandi <sarveshwarb@serverengines.com>
Date: Thu, 27 May 2010 16:28:15 -0700
Subject: [PATCH] be2net: Patch removes redundant while statement in loop.
Git-commit: 84e5b9f75b48fe4a1e4ee72698230701439d0805
Patch-mainline: v2.6.35-rc1
References: FATE#311448, bnc#697255

Signed-off-by: Sarveshwar Bandi <sarveshwarb@serverengines.com>
Signed-off-by: David S. Miller <davem@davemloft.net>

Acked-by: Jeff Mahoney <jeffm@suse.com>
---
 drivers/net/benet/be_main.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)
--- a/drivers/net/benet/be_main.c
+++ b/drivers/net/benet/be_main.c
@@ -1862,7 +1862,7 @@ static int be_setup(struct be_adapter *a
 				goto if_destroy;
 			}
 			vf++;
-		} while (vf < num_vfs);
+		}
 	} else if (!be_physfn(adapter)) {
 		status = be_cmd_mac_addr_query(adapter, mac,
 			MAC_ADDRESS_TYPE_NETWORK, false, adapter->if_handle);
