From u.kleine-koenig@pengutronix.de  Thu Jun 23 15:40:24 2011
From: =?UTF-8?q?Uwe=20Kleine-K=C3=B6nig?= <u.kleine-koenig@pengutronix.de>
Date: Fri, 11 Jun 2010 12:17:01 +0200
Subject: [PATCH] fix typos concerning "management"
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
Git-commit: 65155b3708137fabee865dc4da822763c0c41208
Patch-mainline: v2.6.36-rc1
References: FATE#311448, bnc#697255

Signed-off-by: Uwe Kleine-König <u.kleine-koenig@pengutronix.de>
Signed-off-by: Jiri Kosina <jkosina@suse.cz>

Acked-by: Jeff Mahoney <jeffm@suse.com>
---
 drivers/net/benet/be_hw.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)
--- a/drivers/net/benet/be_hw.h
+++ b/drivers/net/benet/be_hw.h
@@ -52,7 +52,7 @@
  */
 #define MEMBAR_CTRL_INT_CTRL_HOSTINTR_MASK	(1 << 29) /* bit 29 */
 
-/********* Power managment (WOL) **********/
+/********* Power management (WOL) **********/
 #define PCICFG_PM_CONTROL_OFFSET		0x44
 #define PCICFG_PM_CONTROL_MASK			0x108	/* bits 3 & 8 */
 
