From 5aa3bcb53113e49e0f9421e85950ca6c231a724e Mon Sep 17 00:00:00 2001
References: fate#311461. bnc#698050

From: Dmitry Kravkov <dmitry@broadcom.com>
Date: Tue, 9 Aug 2011 13:04:45 -0700
Subject: [PATCH 38/38] bnx2x: call netdev_alloc_mq
Patch-mainline: no (backport fix only)
Acked-by: mmarek@suse.cz

---
 drivers/net/bnx2x/bnx2x_main.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/net/bnx2x/bnx2x_main.c b/drivers/net/bnx2x/bnx2x_main.c
index 4669679..ef8a9d0 100644
--- a/drivers/net/bnx2x/bnx2x_main.c
+++ b/drivers/net/bnx2x/bnx2x_main.c
@@ -10611,7 +10611,7 @@ static int __devinit bnx2x_init_one(struct pci_dev *pdev,
 	tx_count = MAX_TXQS_PER_COS * max_cos_est + FCOE_PRESENT;
 
 	/* dev zeroed in init_etherdev */
-	dev = alloc_etherdev_mq(sizeof(*bp), rx_count);
+	dev = alloc_etherdev_mq(sizeof(*bp), tx_count);
 	if (!dev) {
 		dev_err(&pdev->dev, "Cannot allocate net device\n");
 		return -ENOMEM;
-- 
1.7.6.178.g55272

