From: Guoqing Jiang <gqjiang@suse.com>
Date: Thu, 2 Jun 2016 23:32:05 -0400
Subject: md: simplify the code with md_kick_rdev_from_array
References: bsc#1003941,FATE#321732
Git-commit: db76767213be6c6db1b1344621511e27e9984fbb
Patch-Mainline: v4.8-rc1

Signed-off-by: Guoqing Jiang <gqjiang@suse.com>
Signed-off-by: Shaohua Li <shli@fb.com>
---
 drivers/md/md.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/md/md.c b/drivers/md/md.c
index 00f987f..01cca77 100644
--- a/drivers/md/md.c
+++ b/drivers/md/md.c
@@ -2530,8 +2530,7 @@ static int add_bound_rdev(struct md_rdev *rdev)
 		if (add_journal)
 			mddev_resume(mddev);
 		if (err) {
-			unbind_rdev_from_array(rdev);
-			export_rdev(rdev);
+			md_kick_rdev_from_array(rdev);
 			return err;
 		}
 	}
-- 
1.8.5.6

