From sathya.perla@emulex.com  Thu Jun 23 15:40:25 2011
From: Sathya Perla <sathya.perla@emulex.com>
Date: Wed, 1 Dec 2010 01:02:28 +0000
Subject: [PATCH] be2net: Fix be_dev_family_check() return value check
Git-commit: 63657b9c319588cd35ed869e19cc6255dbef0d20
Patch-mainline: v2.6.38-rc1
References: FATE#311448, bnc#697255

Signed-off-by: Sathya Perla <sathya.perla@emulex.com>
Signed-off-by: David S. Miller <davem@davemloft.net>

Acked-by: Jeff Mahoney <jeffm@suse.com>
---
 drivers/net/benet/be_main.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)
--- a/drivers/net/benet/be_main.c
+++ b/drivers/net/benet/be_main.c
@@ -2806,7 +2806,7 @@ static int __devinit be_probe(struct pci
 	pci_set_drvdata(pdev, adapter);
 
 	status = be_dev_family_check(adapter);
-	if (!status)
+	if (status)
 		goto free_netdev;
 
 	adapter->netdev = netdev;
