From: Chumbalkar, Nagananda <Nagananda.Chumbalkar@hp.com>
Subject: [CPUFREQ] pcc-cpufreq: remove duplicate statements
References: none
Patch-Mainline: v2.6.39-rc1
Git-commit: bdce2595a2f539c6fdedd8f2bd281326b627bba3


Signed-off-by: Thomas Renninger <trenn@suse.de>

Remove a couple of assigment statements that appear twice.

Signed-off-by: Naga Chumbalkar <nagananda.chumbalkar@hp.com>
Signed-off-by: Dave Jones <davej@redhat.com>

diff --git a/arch/x86/kernel/cpu/cpufreq/pcc-cpufreq.c b/arch/x86/kernel/cpu/cpufreq/pcc-cpufreq.c
index 4a5a42b..755a31e 100644
--- a/arch/x86/kernel/cpu/cpufreq/pcc-cpufreq.c
+++ b/arch/x86/kernel/cpu/cpufreq/pcc-cpufreq.c
@@ -315,8 +315,6 @@ static int __init pcc_cpufreq_do_osc(acpi_handle *handle)
 
 	input.count = 4;
 	input.pointer = in_params;
-	input.count = 4;
-	input.pointer = in_params;
 	in_params[0].type               = ACPI_TYPE_BUFFER;
 	in_params[0].buffer.length      = 16;
 	in_params[0].buffer.pointer     = OSC_UUID;
