From: Jens Axboe <axboe@fb.com>
Date: Wed, 30 Mar 2016 10:09:35 -0600
Subject: loop: switch to using blk_queue_write_cache()
References: bnc#1003941,FATE#321732
Patch-Mainline: v4.7-rc1
Git-commit: 21d0727f639e4ba2bd194b2eb9b38ac840bbbc87

Signed-off-by: Jens Axboe <axboe@fb.com>
Reviewed-by: Christoph Hellwig <hch@lst.de>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/block/loop.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/block/loop.c b/drivers/block/loop.c
index fdc16c3..3bf6da0 100644
--- a/drivers/block/loop.c
+++ b/drivers/block/loop.c
@@ -952,7 +952,7 @@ static int loop_set_fd(struct loop_device *lo, fmode_t mode,
 	mapping_set_gfp_mask(mapping, lo->old_gfp_mask & ~(__GFP_IO|__GFP_FS));
 
 	if (!(lo_flags & LO_FLAGS_READ_ONLY) && file->f_op->fsync)
-		blk_queue_flush(lo->lo_queue, REQ_FLUSH);
+		blk_queue_write_cache(lo->lo_queue, true, false);
 
 	loop_update_dio(lo);
 	set_capacity(lo->lo_disk, size);
-- 
1.8.5.6

