From: Jens Axboe <axboe@fb.com>
Date: Wed, 30 Mar 2016 10:16:53 -0600
Subject: md: update to using blk_queue_write_cache()
References: bnc#1003941,FATE#321732
Patch-Mainline: v4.7-rc1
Git-commit: 56883a7ec85f5bc2da7eca67905de54a475daebc

Signed-off-by: Jens Axboe <axboe@fb.com>
Reviewed-by: Christoph Hellwig <hch@lst.de>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/md/md.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/md/md.c b/drivers/md/md.c
index 7f64007..f95eebf 100644
--- a/drivers/md/md.c
+++ b/drivers/md/md.c
@@ -5116,7 +5116,7 @@ static int md_alloc(dev_t dev, char *name)
 	disk->fops = &md_fops;
 	disk->private_data = mddev;
 	disk->queue = mddev->queue;
-	blk_queue_flush(mddev->queue, REQ_FLUSH | REQ_FUA);
+	blk_queue_write_cache(mddev->queue, true, true);
 	/* Allow extended partitions.  This makes the
 	 * 'mdp' device redundant, but we can't really
 	 * remove it now.
-- 
1.8.5.6

