From ajit.khaparde@emulex.com  Thu Jun 23 15:40:25 2011
From: Ajit Khaparde <ajit.khaparde@emulex.com>
Date: Fri, 11 Feb 2011 13:37:25 +0000
Subject: [PATCH] be2net: Fix broken priority setting when vlan tagging is enabled.
Git-commit: 60964dd7087fa252c16022a7590e385294c5472a
Patch-mainline: v2.6.39-rc1
References: FATE#311448, bnc#697255

Signed-off-by: Ajit Khaparde <ajit.khaparde@emulex.com>
Signed-off-by: David S. Miller <davem@davemloft.net>

Acked-by: Jeff Mahoney <jeffm@suse.com>
---
 drivers/net/benet/be_cmds.c |    1 +
 1 file changed, 1 insertion(+)
--- a/drivers/net/benet/be_cmds.c
+++ b/drivers/net/benet/be_cmds.c
@@ -102,6 +102,7 @@ static void be_async_grp5_cos_priority_p
 {
 	if (evt->valid) {
 		adapter->vlan_prio_bmap = evt->available_priority_bmap;
+		adapter->recommended_prio &= ~VLAN_PRIO_MASK;
 		adapter->recommended_prio =
 			evt->reco_default_priority << VLAN_PRIO_SHIFT;
 	}
