From: Jens Axboe <axboe@fb.com>
Date: Wed, 30 Mar 2016 10:10:53 -0600
Subject: nbd: switch to using blk_queue_write_cache()
References: bnc#1003941,FATE#321732
Patch-Mainline: v4.7-rc1
Git-commit: aafb1eecbb79ac135fa8028f961596333ecbb80b

Signed-off-by: Jens Axboe <axboe@fb.com>
Reviewed-by: Christoph Hellwig <hch@lst.de>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/block/nbd.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/block/nbd.c b/drivers/block/nbd.c
index aeafaac..0f8fe76 100644
--- a/drivers/block/nbd.c
+++ b/drivers/block/nbd.c
@@ -751,9 +751,9 @@ static int __nbd_ioctl(struct block_device *bdev, struct nbd_device *nbd,
 			queue_flag_set_unlocked(QUEUE_FLAG_DISCARD,
 				nbd->disk->queue);
 		if (nbd->flags & NBD_FLAG_SEND_FLUSH)
-			blk_queue_flush(nbd->disk->queue, REQ_FLUSH);
+			blk_queue_write_cache(nbd->disk->queue, true, false);
 		else
-			blk_queue_flush(nbd->disk->queue, 0);
+			blk_queue_write_cache(nbd->disk->queue, false, false);
 
 		thread = kthread_run(nbd_thread_send, nbd, "%s",
 				     nbd_name(nbd));
-- 
1.8.5.6

