From: Jon Derrick <jonathan.derrick@intel.com>
Date: Tue, 8 Mar 2016 10:34:54 -0700
Subject: NVMe: Remove unused sq_head read in completion path
Git-commit: 48c7823f42da2bc881ae2e325ed40123871c2fb9
Patch-mainline: v4.6-rc1
References: bsc#964944,FATE#319965

Signed-off-by: Jon Derrick <jonathan.derrick@intel.com>
Reviewed-by: Sagi Grimberg <sagig@mellanox.com>
Signed-off-by: Jens Axboe <axboe@fb.com>
Acked-by: Johannes Thumshirn <jthumshirn@suse.com>
---
 drivers/nvme/host/pci.c |    2 --
 1 file changed, 2 deletions(-)

--- a/drivers/nvme/host/pci.c
+++ b/drivers/nvme/host/pci.c
@@ -134,7 +134,6 @@ struct nvme_queue {
 	u32 __iomem *q_db;
 	u16 q_depth;
 	s16 cq_vector;
-	u16 sq_head;
 	u16 sq_tail;
 	u16 cq_head;
 	u16 qid;
@@ -738,7 +737,6 @@ static void __nvme_process_cq(struct nvm
 
 		if ((status & 1) != phase)
 			break;
-		nvmeq->sq_head = le16_to_cpu(cqe.sq_head);
 		if (++head == nvmeq->q_depth) {
 			head = 0;
 			phase = !phase;
