From sathya.perla@emulex.com  Thu Jun 23 15:40:25 2011
From: Sathya Perla <sathya.perla@emulex.com>
Date: Mon, 21 Mar 2011 20:49:29 +0000
Subject: [PATCH] be2net: remove one useless line
Git-commit: e8c37c80006c99a00aa70a783023d616c166a04b
Patch-mainline: v3.0-rc1
References: FATE#311448, bnc#697255

Signed-off-by: Sathya Perla <sathya.perla@emulex.com>
Signed-off-by: David S. Miller <davem@davemloft.net>

Acked-by: Jeff Mahoney <jeffm@suse.com>
---
 drivers/net/benet/be_main.c |    1 -
 1 file changed, 1 deletion(-)
--- a/drivers/net/benet/be_main.c
+++ b/drivers/net/benet/be_main.c
@@ -2275,7 +2275,6 @@ static int be_clear(struct be_adapter *a
 	if (be_physfn(adapter) && adapter->sriov_enabled)
 		be_vf_eth_addr_rem(adapter);
 
-	be_mcc_queues_destroy(adapter);
 	be_rx_queues_destroy(adapter);
 	be_tx_queues_destroy(adapter);
 
