From: Stanislaw Gruszka <sgruszka@redhat.com>
Date: Sun, 27 Jun 2010 23:26:23 +0000
Subject: e1000e: fail when try to setup unsupported features
References: bnc#699092, fate#311825
Patch-mainline: v2.6.36
Git-commit: 52b6dcfe59d73347a598ba0826a6191a1e497679

Signed-off-by: Stanislaw Gruszka <sgruszka@redhat.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Cho, Yu-Chen <acho@suse.com>
---
 drivers/net/e1000e/ethtool.c |    1 -
 1 file changed, 1 deletion(-)

--- a/drivers/net/e1000e/ethtool.c
+++ b/drivers/net/e1000e/ethtool.c
@@ -2049,7 +2049,6 @@ static const struct ethtool_ops e1000_et
 	.get_coalesce		= e1000_get_coalesce,
 	.set_coalesce		= e1000_set_coalesce,
 	.get_flags		= ethtool_op_get_flags,
-	.set_flags		= ethtool_op_set_flags,
 };
 
 void e1000e_set_ethtool_ops(struct net_device *netdev)
