From: Emil Tantilov <emil.s.tantilov@intel.com>
Date: Wed, 28 Apr 2010 14:24:51 -0700
Subject: igb: Clean up left over prototype of igb_get_hw_dev_name()
References: FATE#311863 ,bnc#699089
Patch-mainline: v2.6.35-rc1
Git-commit: cfc1fbb079b265bf69d4ceba590a2e2c1a1cde33

Signed-off-by: Emil Tantilov <emil.s.tantilov@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/igb/igb.h |    1 -
 1 file changed, 1 deletion(-)

--- a/drivers/net/igb/igb.h
+++ b/drivers/net/igb/igb.h
@@ -338,7 +338,6 @@ enum igb_boards {
 extern char igb_driver_name[];
 extern char igb_driver_version[];
 
-extern char *igb_get_hw_dev_name(struct e1000_hw *hw);
 extern int igb_up(struct igb_adapter *);
 extern void igb_down(struct igb_adapter *);
 extern void igb_reinit_locked(struct igb_adapter *);
