From davem@davemloft.net  Thu Jun 23 15:40:26 2011
From: David S. Miller <davem@davemloft.net>
Date: Thu, 19 May 2011 18:49:54 -0400
Subject: [PATCH] be2net: Kill set but unused variable 'req' in lancer_fw_download()
Git-commit: 63722966d703aeb8071d52172de87e377006cd28
Patch-mainline: v3.0-rc1
References: FATE#311448, bnc#697255

Signed-off-by: David S. Miller <davem@davemloft.net>

Acked-by: Jeff Mahoney <jeffm@suse.com>
---
 drivers/net/benet/be_main.c |    2 --
 1 file changed, 2 deletions(-)
--- a/drivers/net/benet/be_main.c
+++ b/drivers/net/benet/be_main.c
@@ -2647,7 +2647,6 @@ static int lancer_fw_download(struct be_
 #define LANCER_FW_DOWNLOAD_CHUNK      (32 * 1024)
 #define LANCER_FW_DOWNLOAD_LOCATION   "/prg"
 	struct be_dma_mem flash_cmd;
-	struct lancer_cmd_req_write_object *req;
 	const u8 *data_ptr = NULL;
 	u8 *dest_image_ptr = NULL;
 	size_t image_size = 0;
@@ -2676,7 +2675,6 @@ static int lancer_fw_download(struct be_
 		goto lancer_fw_exit;
 	}
 
-	req = flash_cmd.va;
 	dest_image_ptr = flash_cmd.va +
 				sizeof(struct lancer_cmd_req_write_object);
 	image_size = fw->size;
