From: Joe Perches <joe@perches.com>
Date: Sun, 14 Nov 2010 17:04:32 +0000
Subject: drivers/net/e1000e: Remove unnecessary semicolons
References: bnc#699092, fate#311825
Patch-mainline: v2.6.38
Git-commit: 1d51c4185bb369cb39ed8cc20b331508e47b35b2

Signed-off-by: Joe Perches <joe@perches.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Cho, Yu-Chen <acho@suse.com>
---
 drivers/net/e1000e/netdev.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/e1000e/netdev.c
+++ b/drivers/net/e1000e/netdev.c
@@ -4511,7 +4511,7 @@ dma_error:
 			i += tx_ring->count;
 		i--;
 		buffer_info = &tx_ring->buffer_info[i];
-		e1000_put_txbuf(adapter, buffer_info);;
+		e1000_put_txbuf(adapter, buffer_info);
 	}
 
 	return 0;
