From: Carolyn Wyborny <carolyn.wyborny@intel.com>
Date: Tue, 12 Oct 2010 22:27:02 +0000
Subject: igb: add check for fiber/serdes devices to igb_set_spd_dplx;
References: FATE#311863 ,bnc#699089
Patch-mainline: v2.6.37-rc1
Git-commit: cd2638a86c7b90e77ce623c09de2a26177f2a5c1

Signed-off-by: Carolyn Wyborny <carolyn.wyborny@intel.com>
Tested-by: Emil Tantilov <emil.s.tantilov@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/igb/igb_main.c |    7 +++++++
 1 file changed, 7 insertions(+)

--- a/drivers/net/igb/igb_main.c
+++ b/drivers/net/igb/igb_main.c
@@ -6062,6 +6062,13 @@ int igb_set_spd_dplx(struct igb_adapter
 
 	mac->autoneg = 0;
 
+	/* Fiber NIC's only allow 1000 Gbps Full duplex */
+	if ((adapter->hw.phy.media_type == e1000_media_type_internal_serdes) &&
+		spddplx != (SPEED_1000 + DUPLEX_FULL)) {
+		dev_err(&pdev->dev, "Unsupported Speed/Duplex configuration\n");
+		return -EINVAL;
+	}
+
 	switch (spddplx) {
 	case SPEED_10 + DUPLEX_HALF:
 		mac->forced_speed_duplex = ADVERTISE_10_HALF;
