From sony.chacko@qlogic.com  Thu Jun 23 22:49:06 2011
From: Sony Chacko <sony.chacko@qlogic.com>
Date: Mon, 31 Jan 2011 06:56:29 -0800
Subject: [PATCH 175/187] qlcnic: update driver version to 5.0.15.2
Patch-mainline: Never
References: bnc#698272, FATE#311468

Signed-off-by: Sony Chacko <sony.chacko@qlogic.com>
Acked-by: Jeff Mahoney <jeffm@suse.com>
---
 drivers/net/qlcnic/qlcnic.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/net/qlcnic/qlcnic.h b/drivers/net/qlcnic/qlcnic.h
index 7e458ad..b983b17 100644
--- a/drivers/net/qlcnic/qlcnic.h
+++ b/drivers/net/qlcnic/qlcnic.h
@@ -36,7 +36,7 @@
 #define _QLCNIC_LINUX_MAJOR 5
 #define _QLCNIC_LINUX_MINOR 0
 #define _QLCNIC_LINUX_SUBVERSION 15
-#define QLCNIC_LINUX_VERSIONID  "5.0.15.1"
+#define QLCNIC_LINUX_VERSIONID  "5.0.15.2"
 #define QLCNIC_DRV_IDC_VER  0x01
 #define QLCNIC_DRIVER_VERSION  ((_QLCNIC_LINUX_MAJOR << 16) |\
 		 (_QLCNIC_LINUX_MINOR << 8) | (_QLCNIC_LINUX_SUBVERSION))
-- 
1.6.3.3

