From: Christoph Hellwig <hch@lst.de>
Date: Sun, 11 Sep 2016 16:03:02 +0200
Subject: block: remove bio_destructor_t
References: bnc#1023798,FATE#321463
Patch-Mainline: v4.9-rc1
Git-commit: 3f7c624aa58f769e0313ca3310704c5d88ac99ce

Signed-off-by: Christoph Hellwig <hch@lst.de>
Reviewed-by: Bart Van Assche <bart.vanassche@sandisk.com>
Signed-off-by: Jens Axboe <axboe@fb.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 include/linux/blk_types.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/include/linux/blk_types.h b/include/linux/blk_types.h
index 53ee1a2..cd395ec 100644
--- a/include/linux/blk_types.h
+++ b/include/linux/blk_types.h
@@ -16,7 +16,6 @@ struct block_device;
 struct io_context;
 struct cgroup_subsys_state;
 typedef void (bio_end_io_t) (struct bio *);
-typedef void (bio_destructor_t) (struct bio *);
 
 #ifdef CONFIG_BLOCK
 /*
-- 
1.8.5.6

