From: Christoph Hellwig <hch@lst.de>
Date: Sun, 11 Sep 2016 16:03:03 +0200
Subject: bio.h: remove a very outdated comment
References: bnc#1023798,FATE#321463
Patch-Mainline: v4.9-rc1
Git-commit: fc95db3edeaf924e9ad16592d9c1b06c730a49c9

Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Jens Axboe <axboe@fb.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 include/linux/bio.h | 2 --
 1 file changed, 2 deletions(-)

diff --git a/include/linux/bio.h b/include/linux/bio.h
index 23ddf4b..e00721a 100644
--- a/include/linux/bio.h
+++ b/include/linux/bio.h
@@ -1,6 +1,4 @@
 /*
- * 2.5 block I/O model
- *
  * Copyright (C) 2001 Jens Axboe <axboe@suse.de>
  *
  * This program is free software; you can redistribute it and/or modify
-- 
1.8.5.6

