From: Christoph Hellwig <hch@lst.de>
Date: Sun, 11 Sep 2016 16:03:04 +0200
Subject: block: remove IOPRIO_BITS
References: bnc#1023798,FATE#321463
Patch-Mainline: v4.9-rc1
Git-commit: c5c5ca777469f0ff854f1da0aff9b3a9051b3ef7

Signed-off-by: Christoph Hellwig <hch@lst.de>
Reviewed-by: Bart Van Assche <bart.vanassche@sandisk.com>
Signed-off-by: Jens Axboe <axboe@fb.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 include/linux/ioprio.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/include/linux/ioprio.h b/include/linux/ioprio.h
index beb9ce1..8c12390 100644
--- a/include/linux/ioprio.h
+++ b/include/linux/ioprio.h
@@ -7,7 +7,6 @@
 /*
  * Gives us 8 prio classes with 13-bits of data for each class
  */
-#define IOPRIO_BITS		(16)
 #define IOPRIO_CLASS_SHIFT	(13)
 #define IOPRIO_PRIO_MASK	((1UL << IOPRIO_CLASS_SHIFT) - 1)
 
-- 
1.8.5.6

