From 7157c2151b870639e01585941d2aa7980a3d486a Mon Sep 17 00:00:00 2001
From: James Smart <jsmart2021@gmail.com>
Date: Fri, 2 Dec 2016 00:28:39 -0800
Subject: nvme-fabrics: Add FC transport error codes to nvme.h
References: bsc#993388, FATE#321732
Git-commit: cba3bdfd2e89edd706e2c40dfad914aca663b6ac
Patch-mainline: v4.10-rc1

Signed-off-by: James Smart <james.smart@broadcom.com>
Reviewed-by: Christoph Hellwig <hch@lst.de>
Reviewed-by: Jay Freyensee <james_p_freyensee@linux.intel.com>
Reviewed-by: Johannes Thumshirn <jthumshirn@suse.de>
Signed-off-by: Christoph Hellwig <hch@lst.de>
Signed-off-by: Johannes Thumshirn <jthumshirn@suse.de>

---
 include/linux/nvme.h | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/include/linux/nvme.h b/include/linux/nvme.h
index 81e6241..bedd86f 100644
--- a/include/linux/nvme.h
+++ b/include/linux/nvme.h
@@ -974,6 +974,19 @@ enum {
 	NVME_SC_UNWRITTEN_BLOCK		= 0x287,
 
 	NVME_SC_DNR			= 0x4000,
+
+
+	/*
+	 * FC Transport-specific error status values for NVME commands
+	 *
+	 * Transport-specific status code values must be in the range 0xB0..0xBF
+	 */
+
+	/* Generic FC failure - catchall */
+	NVME_SC_FC_TRANSPORT_ERROR	= 0x00B0,
+
+	/* I/O failure due to FC ABTS'd */
+	NVME_SC_FC_TRANSPORT_ABORTED	= 0x00B1,
 };
 
 struct nvme_completion {
-- 
1.8.5.6

