From: Jens Axboe <axboe@fb.com>
Date: Fri, 2 Dec 2016 20:00:14 -0700
Subject: blk-mq: blk_account_io_start() takes a bool
References: bsc#1020989,FATE#322379
Git-commit: 6e85eaf3078bcc8552ca32a0938dbf7d2b495af0
Patch-Mainline: v4.10-rc1

Signed-off-by: Jens Axboe <axboe@fb.com>
Reviewed-by: Christoph Hellwig <hch@lst.de>
Reviewed-by: Johannes Thumshirn <jthumshirn@suse.de>
Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 block/blk-mq.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/block/blk-mq.c b/block/blk-mq.c
index 578b056..ba05753 100644
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -1237,7 +1237,7 @@ static void blk_mq_bio_to_request(struct request *rq, struct bio *bio)
 {
 	init_request_from_bio(rq, bio);
 
-	blk_account_io_start(rq, 1);
+	blk_account_io_start(rq, true);
 }
 
 static inline bool hctx_allow_merges(struct blk_mq_hw_ctx *hctx)
-- 
1.8.5.6

