From: Ritesh Harjani <riteshh@codeaurora.org>
Date: Fri, 16 Dec 2016 10:11:56 +0530
Subject: block: Remove unused member (busy) from struct blk_queue_tag
References: bsc#1020989,FATE#322379
Git-commit: e8465447d2f3366069115f7453153561ac9a1220
Patch-Mainline: v4.10-rc1

Signed-off-by: Ritesh Harjani <riteshh@codeaurora.org>
Reviewed-by: Bart Van Assche <bart.vanassche@sandisk.com>
Signed-off-by: Jens Axboe <axboe@fb.com>
Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 include/linux/blkdev.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/include/linux/blkdev.h b/include/linux/blkdev.h
index 9f7a799..317b9cf1 100644
--- a/include/linux/blkdev.h
+++ b/include/linux/blkdev.h
@@ -288,7 +288,6 @@ enum blk_queue_state {
 struct blk_queue_tag {
 	struct request **tag_index;	/* map of busy tags */
 	unsigned long *tag_map;		/* bit map of free/busy tags */
-	int busy;			/* current depth */
 	int max_depth;			/* what we will send to device */
 	int real_max_depth;		/* what the array can hold */
 	atomic_t refcnt;		/* map can be shared */
-- 
1.8.5.6

