From: Borislav Petkov <borislav.petkov@amd.com>
Subject: amd64_edac: remove superfluous dbg printk
References: fate#311968
Patch-Mainline: yes
Git-commit: ad858bfa14277a22a343bb89e99c9af0a97fc456


Signed-off-by: Thomas Renninger <trenn@suse.de>

Signed-off-by: Borislav Petkov <borislav.petkov@amd.com>

---
 drivers/edac/amd64_edac.c |    3 ---
 1 file changed, 3 deletions(-)

Index: linux-2.6.32-SLE11-SP1/drivers/edac/amd64_edac.c
===================================================================
--- linux-2.6.32-SLE11-SP1.orig/drivers/edac/amd64_edac.c
+++ linux-2.6.32-SLE11-SP1/drivers/edac/amd64_edac.c
@@ -1338,9 +1338,6 @@ static void f10_read_dram_base_limit(str
 	/* Read from the ECS data register for the HIGH portion */
 	amd64_read_pci_cfg(pvt->addr_f1_ctl, high_offset, &high_limit);
 
-	debugf0("  HW Regs: BASE=0x%08x-%08x      LIMIT=  0x%08x-%08x\n",
-		high_base, low_base, high_limit, low_limit);
-
 	pvt->dram_DstNode[dram] = (low_limit & 0x7);
 	pvt->dram_IntlvSel[dram] = (low_limit >> 8) & 0x7;
 
