From: Mauro Carvalho Chehab <mchehab@redhat.com>
Subject: i7core_edac: Some cleanups at displayed info
References: fate#311968
Git-commit: 17cb7b0cf78c14913c5410eff2ce03e1d9c8d958
Patch-mainline: v2.6.35-rc2


Signed-off-by: Thomas Renninger <trenn@suse.de>

Signed-off-by: Mauro Carvalho Chehab <mchehab@redhat.com>

---
 drivers/edac/i7core_edac.c |   23 ++++++++++-------------
 1 file changed, 10 insertions(+), 13 deletions(-)

Index: linux-2.6.32-SLE11-SP1/drivers/edac/i7core_edac.c
===================================================================
--- linux-2.6.32-SLE11-SP1.orig/drivers/edac/i7core_edac.c
+++ linux-2.6.32-SLE11-SP1/drivers/edac/i7core_edac.c
@@ -463,8 +463,8 @@ static int get_dimm_config(struct mem_ct
 	pci_read_config_dword(pdev, MC_MAX_DOD, &pvt->info.max_dod);
 	pci_read_config_dword(pdev, MC_CHANNEL_MAPPER, &pvt->info.ch_map);
 
-	debugf0("MC control=0x%08x status=0x%08x dod=0x%08x map=0x%08x\n",
-		pvt->info.mc_control, pvt->info.mc_status,
+	debugf0("QPI %d control=0x%08x status=0x%08x dod=0x%08x map=0x%08x\n",
+		socket, pvt->info.mc_control, pvt->info.mc_status,
 		pvt->info.max_dod, pvt->info.ch_map);
 
 	if (ECC_ENABLED(pvt)) {
@@ -479,16 +479,14 @@ static int get_dimm_config(struct mem_ct
 	}
 
 	/* FIXME: need to handle the error codes */
-	debugf0("DOD Max limits: DIMMS: %d, %d-ranked, %d-banked\n",
+	debugf0("DOD Max limits: DIMMS: %d, %d-ranked, %d-banked "
+		"x%x x 0x%x\n",
 		numdimms(pvt->info.max_dod),
 		numrank(pvt->info.max_dod >> 2),
-		numbank(pvt->info.max_dod >> 4));
-	debugf0("DOD Max rows x colums = 0x%x x 0x%x\n",
+		numbank(pvt->info.max_dod >> 4),
 		numrow(pvt->info.max_dod >> 6),
 		numcol(pvt->info.max_dod >> 9));
 
-	debugf0("Memory channel configuration:\n");
-
 	for (i = 0; i < NUM_CHANS; i++) {
 		u32 data, dimm_dod[3], value[8];
 
@@ -554,10 +552,9 @@ static int get_dimm_config(struct mem_ct
 
 			pvt->channel[socket][i].dimms++;
 
-			debugf0("\tdimm %d (0x%08x) %d Mb offset: %x, "
-				"numbank: %d,\n\t\t"
-				"numrank: %d, numrow: %#x, numcol: %#x\n",
-				j, dimm_dod[j], size,
+			debugf0("\tdimm %d %d Mb offset: %x, "
+				"bank: %d, rank: %d, row: %#x, col: %#x\n",
+				j, size,
 				RANKOFFSET(dimm_dod[j]),
 				banks, ranks, rows, cols);
 
@@ -609,9 +606,9 @@ static int get_dimm_config(struct mem_ct
 		pci_read_config_dword(pdev, MC_SAG_CH_5, &value[5]);
 		pci_read_config_dword(pdev, MC_SAG_CH_6, &value[6]);
 		pci_read_config_dword(pdev, MC_SAG_CH_7, &value[7]);
-		debugf0("\t[%i] DIVBY3\tREMOVED\tOFFSET\n", i);
+		debugf1("\t[%i] DIVBY3\tREMOVED\tOFFSET\n", i);
 		for (j = 0; j < 8; j++)
-			debugf0("\t\t%#x\t%#x\t%#x\n",
+			debugf1("\t\t%#x\t%#x\t%#x\n",
 				(value[j] >> 27) & 0x1,
 				(value[j] >> 24) & 0x7,
 				(value[j] && ((1 << 24) - 1)));
