From: Mauro Carvalho Chehab <mchehab@redhat.com>
Subject: i7core_edac: Improve debug to seek for register/remove errors
References: fate#311968
Git-commit: 3cfd01468b98360ede8cc8849d14e586253d290c
Patch-mainline: v2.6.37-rc1


Signed-off-by: Thomas Renninger <trenn@suse.de>

Signed-off-by: Mauro Carvalho Chehab <mchehab@redhat.com>

---
 drivers/edac/i7core_edac.c |   11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

Index: linux-2.6.32-SLE11-SP1/drivers/edac/i7core_edac.c
===================================================================
--- linux-2.6.32-SLE11-SP1.orig/drivers/edac/i7core_edac.c
+++ linux-2.6.32-SLE11-SP1/drivers/edac/i7core_edac.c
@@ -1890,7 +1890,8 @@ static int i7core_register_mci(struct i7
 	if (unlikely(!mci))
 		return -ENOMEM;
 
-	debugf0("MC: " __FILE__ ": %s(): mci = %p\n", __func__, mci);
+	debugf0("MC: " __FILE__ ": %s(): mci = %p, dev = %p\n",
+		__func__, mci, &i7core_dev->pdev[0]->dev);
 
 	/* record ptr to the generic device */
 	mci->dev = &i7core_dev->pdev[0]->dev;
@@ -2058,12 +2059,18 @@ static void __devexit i7core_remove(stru
 	list_for_each_entry_safe(i7core_dev, tmp, &i7core_edac_list, list) {
 		mci = find_mci_by_dev(&i7core_dev->pdev[0]->dev);
 		if (unlikely(!mci || !mci->pvt_info)) {
-			i7core_printk(KERN_ERR,
+			debugf0("MC: " __FILE__ ": %s(): dev = %p\n",
+				__func__, &i7core_dev->pdev[0]->dev);
+
+				i7core_printk(KERN_ERR,
 				      "Couldn't find mci hanler\n");
 		} else {
 			pvt = mci->pvt_info;
 			i7core_dev = pvt->i7core_dev;
 
+			debugf0("MC: " __FILE__ ": %s(): mci = %p, dev = %p\n",
+				__func__, mci, &i7core_dev->pdev[0]->dev);
+
 			if (likely(pvt->i7core_pci))
 				edac_pci_release_generic_ctl(pvt->i7core_pci);
 			else
