From: Mauro Carvalho Chehab <mchehab@redhat.com>
Subject: i7core_edac: fix error codes for sysfs error injection interface
References: fate#311968
Git-commit: 2068def56c09f2b24e6de04a1b84757a0fb07947
Patch-mainline: v2.6.35-rc2


Signed-off-by: Thomas Renninger <trenn@suse.de>

Signed-off-by: Mauro Carvalho Chehab <mchehab@redhat.com>

---
 drivers/edac/i7core_edac.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

Index: linux-2.6.32-SLE11-SP1/drivers/edac/i7core_edac.c
===================================================================
--- linux-2.6.32-SLE11-SP1.orig/drivers/edac/i7core_edac.c
+++ linux-2.6.32-SLE11-SP1/drivers/edac/i7core_edac.c
@@ -645,7 +645,7 @@ static ssize_t i7core_inject_socket_stor
 
 	rc = strict_strtoul(data, 10, &value);
 	if ((rc < 0) || (value >= pvt->sockets))
-		return 0;
+		return -EIO;
 
 	pvt->inject.section = (u32) value;
 	return count;
@@ -677,7 +677,7 @@ static ssize_t i7core_inject_section_sto
 
 	rc = strict_strtoul(data, 10, &value);
 	if ((rc < 0) || (value > 3))
-		return 0;
+		return -EIO;
 
 	pvt->inject.section = (u32) value;
 	return count;
@@ -710,7 +710,7 @@ static ssize_t i7core_inject_type_store(
 
 	rc = strict_strtoul(data, 10, &value);
 	if ((rc < 0) || (value > 7))
-		return 0;
+		return -EIO;
 
 	pvt->inject.type = (u32) value;
 	return count;
@@ -745,7 +745,7 @@ static ssize_t i7core_inject_eccmask_sto
 
 	rc = strict_strtoul(data, 10, &value);
 	if (rc < 0)
-		return 0;
+		return -EIO;
 
 	pvt->inject.eccmask = (u32) value;
 	return count;
