From: Mauro Carvalho Chehab <mchehab@redhat.com>
Subject: i7core_edac: We need to use list_for_each_entry_safe to avoid errors
References: fate#311968
Git-commit: 4253868034221db6e42dbbb61e0305fe1757f8da
Patch-mainline: v2.6.35-rc2


Signed-off-by: Thomas Renninger <trenn@suse.de>

Signed-off-by: Mauro Carvalho Chehab <mchehab@redhat.com>

---
 drivers/edac/i7core_edac.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

Index: linux-2.6.32-SLE11-SP1/drivers/edac/i7core_edac.c
===================================================================
--- linux-2.6.32-SLE11-SP1.orig/drivers/edac/i7core_edac.c
+++ linux-2.6.32-SLE11-SP1/drivers/edac/i7core_edac.c
@@ -1125,6 +1125,7 @@ static struct mcidev_sysfs_attribute i7c
 		.show  = i7core_ce_regs_show,
 		.store = NULL,
 	},
+	{ .attr = { .name = NULL } }
 };
 
 /****************************************************************************
@@ -1156,9 +1157,9 @@ static void i7core_put_devices(struct i7
 
 static void i7core_put_all_devices(void)
 {
-	struct i7core_dev *i7core_dev;
+	struct i7core_dev *i7core_dev, *tmp;
 
-	list_for_each_entry(i7core_dev, &i7core_edac_list, list)
+	list_for_each_entry_safe(i7core_dev, tmp, &i7core_edac_list, list)
 		i7core_put_devices(i7core_dev);
 }
 
