From: Stephen Rothwell <sfr@canb.auug.org.au>
Subject: i7core_edac: do not export static functions
References: fate#311968
Git-commit: 4f7b9e7cbe68c97dbe1266709ecfc8b807b0d0ee
Patch-mainline: v2.6.35-rc2


Signed-off-by: Thomas Renninger <trenn@suse.de>

Signed-off-by: Stephen Rothwell <sfr@canb.auug.org.au>
Signed-off-by: Mauro Carvalho Chehab <mchehab@redhat.com>

---
 arch/x86/pci/legacy.c |    1 -
 1 file changed, 1 deletion(-)

Index: linux-2.6.32-SLE11-SP1/arch/x86/pci/legacy.c
===================================================================
 arch/x86/pci/legacy.c |    1 -
 1 file changed, 1 deletion(-)

--- linux-2.6.32-SLE11-SP1.orig/arch/x86/pci/legacy.c
+++ linux-2.6.32-SLE11-SP1/arch/x86/pci/legacy.c
@@ -38,7 +38,6 @@ static int __init pci_legacy_init(void)
 
 	return 0;
 }
-EXPORT_SYMBOL_GPL(pci_legacy_init);
 
 void pcibios_scan_specific_bus(int busn)
 {
