From: Mauro Carvalho Chehab <mchehab@redhat.com>
Subject: i7core_edac: Avoid printing a warning when debug is disabled
References: fate#311968
Patch-Mainline: yes
Git-commit: 4af91889e02c9933823ca8c62fc6f05dfd15f3bd


Signed-off-by: Thomas Renninger <trenn@suse.de>

Signed-off-by: Mauro Carvalho Chehab <mchehab@redhat.com>

---
 drivers/edac/i7core_edac.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

Index: linux-2.6.32-SLE11-SP1/drivers/edac/i7core_edac.c
===================================================================
--- linux-2.6.32-SLE11-SP1.orig/drivers/edac/i7core_edac.c
+++ linux-2.6.32-SLE11-SP1/drivers/edac/i7core_edac.c
@@ -472,7 +472,6 @@ static int get_dimm_config(struct mem_ct
 	struct csrow_info *csr;
 	struct pci_dev *pdev;
 	int i, j;
-	u8 socket = pvt->i7core_dev->socket;
 	unsigned long last_page = 0;
 	enum edac_type mode;
 	enum mem_type mtype;
@@ -489,7 +488,7 @@ static int get_dimm_config(struct mem_ct
 	pci_read_config_dword(pdev, MC_CHANNEL_MAPPER, &pvt->info.ch_map);
 
 	debugf0("QPI %d control=0x%08x status=0x%08x dod=0x%08x map=0x%08x\n",
-		socket, pvt->info.mc_control, pvt->info.mc_status,
+		pvt->i7core_dev->socket, pvt->info.mc_control, pvt->info.mc_status,
 		pvt->info.max_dod, pvt->info.ch_map);
 
 	if (ECC_ENABLED(pvt)) {
