From: Thomas Renninger <trenn@suse.de>
Subject: Enable HEST parsing on Dell platforms by default
References: bnc#714106
Patch-Mainline: never

---
 drivers/acpi/blacklist.c |    7 +++++++
 1 file changed, 7 insertions(+)

Index: linux-3.0-SLE11-SP2-3.0/drivers/acpi/blacklist.c
===================================================================
--- linux-3.0-SLE11-SP2-3.0.orig/drivers/acpi/blacklist.c
+++ linux-3.0-SLE11-SP2-3.0/drivers/acpi/blacklist.c
@@ -157,6 +157,13 @@ static void __init acpi_apei_blacklist(v
 		return;
 
 	acpi_apei_disable();
+
+	if (dmi_name_in_vendors("Dell")) {
+		printk (KERN_INFO "Dell detected: Enabling APEI HEST parsing"
+			" if available\n");
+		hest_disable = 0;
+	}
+
 }
 #else /* !CONFIG_ACPI_APEI */
 #define acpi_apei_blacklist() ((void)0)
