From: Sander Eikelenboom <linux@eikelenboom.it>
Date: Sun, 20 Mar 2016 22:27:06 +0100
Subject: libata: Fixup awkward whitespace in warning by removing line
References: bsc#977572
Patch-mainline: v4.7-rc1
Git-commit: d991c872ac7ffaacc4df93efbfbcb4189cee6440
 continuation.

Signed-off-by: Sander Eikelenboom <linux@eikelenboom.it>
Signed-off-by: Tejun Heo <tj@kernel.org>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/ata/libahci.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/ata/libahci.c b/drivers/ata/libahci.c
index 3982054..e13ba72 100644
--- a/drivers/ata/libahci.c
+++ b/drivers/ata/libahci.c
@@ -2549,8 +2549,8 @@ int ahci_host_activate(struct ata_host *host, struct scsi_host_template *sht)
 
 	if (hpriv->flags & (AHCI_HFLAG_MULTI_MSI | AHCI_HFLAG_MULTI_MSIX)) {
 		if (hpriv->irq_handler)
-			dev_warn(host->dev, "both AHCI_HFLAG_MULTI_MSI flag set \
-				 and custom irq handler implemented\n");
+			dev_warn(host->dev,
+			         "both AHCI_HFLAG_MULTI_MSI flag set and custom irq handler implemented\n");
 
 		rc = ahci_host_activate_multi_irqs(host, sht);
 	} else {
-- 
1.8.5.6

