From: Subbu Seetharaman <subbus@serverengines.com>
Subject: be2net: update version 2.101.346u to 2.102.147s
Patch-mainline: never, SUSE designated version number
References: bnc#585124

Signed-off-by: Subbu Seetharaman <subbus@serverengines.com>
Signed-off-by: Brandon Philips <bphilips@suse.de>

---

diff -Naur a/drivers/net/benet/be.h b/drivers/net/benet/be.h
--- a/drivers/net/benet/be.h	2010-03-05 06:20:15.000000000 +0530
+++ b/drivers/net/benet/be.h	2010-03-05 06:34:43.000000000 +0530
@@ -32,7 +32,7 @@
 
 #include "be_hw.h"
 
-#define DRV_VER			"2.101.346u"
+#define DRV_VER			"2.102.147s"
 #define DRV_NAME		"be2net"
 #define BE_NAME			"ServerEngines BladeEngine2 10Gbps NIC"
 #define BE3_NAME		"ServerEngines BladeEngine3 10Gbps NIC"
