From: Sarveshwar Bandi <sarveshwarb@serverengines.com>
Date: Tue, 19 Jan 2010 05:15:36 +0000
Subject: [PATCH] be2net: Add link test to list of ethtool self tests.
References: bnc#556959
Patch-mainline: applied net-next-2.6
Git-commit: 4276e47e2d1c85a2477caf0d22b91c4f2377fba8

Signed-off-by: Sarveshwar Bandi <sarveshwarb@serverengines.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Brandon Philips <bphilips@suse.de>

---
 drivers/net/benet/be_ethtool.c |   11 +++++++++++
 1 file changed, 11 insertions(+)

Index: linux-2.6.32-SLE11-SP1/drivers/net/benet/be_ethtool.c
===================================================================
--- linux-2.6.32-SLE11-SP1.orig/drivers/net/benet/be_ethtool.c
+++ linux-2.6.32-SLE11-SP1/drivers/net/benet/be_ethtool.c
@@ -112,6 +112,7 @@ static const char et_self_tests[][ETH_GS
 	"PHY Loopback test",
 	"External Loopback test",
 	"DDR DMA test"
+	"Link test"
 };
 
 #define ETHTOOL_TESTS_NUM ARRAY_SIZE(et_self_tests)
@@ -529,6 +530,9 @@ static void
 be_self_test(struct net_device *netdev, struct ethtool_test *test, u64 *data)
 {
 	struct be_adapter *adapter = netdev_priv(netdev);
+	bool link_up;
+	u8 mac_speed = 0;
+	u16 qos_link_speed = 0;
 
 	memset(data, 0, sizeof(u64) * ETHTOOL_TESTS_NUM);
 
@@ -551,6 +555,13 @@ be_self_test(struct net_device *netdev,
 			test->flags |= ETH_TEST_FL_FAILED;
 	}
 
+	if (be_cmd_link_status_query(adapter, &link_up, &mac_speed,
+				&qos_link_speed) != 0) {
+		test->flags |= ETH_TEST_FL_FAILED;
+		data[4] = -1;
+	} else if (mac_speed) {
+		data[4] = 1;
+	}
 }
 
 static int
