From: Naga Chumbalkar <nagananda.chumbalkar@hp.com>
Subject: cpufreq: pcc-cpufreq: sanity check to prevent a NULL pointer dereference
Patch-Mainline: submitted, aimed for 3.2, should show up in stable
References: bnc#=709412

Signed-off-by: Thomas Renninger <trenn@suse.de>

---
 drivers/cpufreq/pcc-cpufreq.c |    3 +++
 1 file changed, 3 insertions(+)

Index: linux-3.0-master/drivers/cpufreq/pcc-cpufreq.c
===================================================================
--- linux-3.0-master.orig/drivers/cpufreq/pcc-cpufreq.c
+++ linux-3.0-master/drivers/cpufreq/pcc-cpufreq.c
@@ -261,6 +261,9 @@ static int pcc_get_offset(int cpu)
 	pr = per_cpu(processors, cpu);
 	pcc_cpu_data = per_cpu_ptr(pcc_cpu_info, cpu);
 
+	if (!pr)
+		return -ENODEV;
+
 	status = acpi_evaluate_object(pr->handle, "PCCP", NULL, &buffer);
 	if (ACPI_FAILURE(status))
 		return -ENODEV;
