From: Hariprasad Shenai <hariprasad@chelsio.com>
Date: Thu, 17 Dec 2015 13:45:07 +0530
Subject: cxgb4: Use symbolic constant for VLAN priority calculation
Patch-mainline: v4.5-rc1
Git-commit: e41e282428ee897b8817049e6ee0b7d5c9caf005
References: bsc#963896 FATE#320113

Signed-off-by: Hariprasad Shenai <hariprasad@chelsio.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Benjamin Poirier <bpoirier@suse.com>
---
 drivers/net/ethernet/chelsio/cxgb4/l2t.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/ethernet/chelsio/cxgb4/l2t.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/l2t.c
@@ -66,7 +66,7 @@ struct l2t_data {
 
 static inline unsigned int vlan_prio(const struct l2t_entry *e)
 {
-	return e->vlan >> 13;
+	return e->vlan >> VLAN_PRIO_SHIFT;
 }
 
 static inline void l2t_hold(struct l2t_data *d, struct l2t_entry *e)
