From: Heinz Mauelshagen <heinzm@redhat.com>
Date: Wed, 15 Jun 2016 18:43:55 +0200
Subject: dm raid: prohibit to pass in both sync and nosync ctr flags
References: bnc#1023798,FATE#321463
Patch-Mainline: v4.8-rc1
Git-commit: 0d851d14b8dfed601e165ccec2819cab492442d8

Signed-off-by: Heinz Mauelshagen <heinzm@redhat.com>
Signed-off-by: Mike Snitzer <snitzer@redhat.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/md/dm-raid.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/md/dm-raid.c b/drivers/md/dm-raid.c
index 4bf7747..e8e9b6a 100644
--- a/drivers/md/dm-raid.c
+++ b/drivers/md/dm-raid.c
@@ -1297,6 +1297,12 @@ static int parse_raid_params(struct raid_set *rs, struct dm_arg_set *as,
 		}
 	}
 
+	if (test_bit(__CTR_FLAG_SYNC, &rs->ctr_flags) &&
+	    test_bit(__CTR_FLAG_NOSYNC, &rs->ctr_flags)) {
+		rs->ti->error = "sync and nosync are mutually exclusive";
+		return -EINVAL;
+	}
+
 	if (validate_region_size(rs, region_size))
 		return -EINVAL;
 
-- 
1.8.5.6

