From: Egbert Eich <eich@suse.de>
Date: Mon May 5 10:41:15 2014 +0200
Subject: drm/cirrus: Create a sysfs entry for connector
References: bnc#876305
Patch-mainline: to be upstreamed
Signed-off-by: Egbert Eich <eich@suse.com>

Signed-off-by: Egbert Eich <eich@suse.de>
---
 drivers/gpu/drm/cirrus/cirrus_mode.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/cirrus/cirrus_mode.c b/drivers/gpu/drm/cirrus/cirrus_mode.c
index baaeb8c..d7e0a98 100644
--- a/drivers/gpu/drm/cirrus/cirrus_mode.c
+++ b/drivers/gpu/drm/cirrus/cirrus_mode.c
@@ -537,6 +537,7 @@ static enum drm_connector_status cirrus_vga_detect(struct drm_connector
 
 static void cirrus_connector_destroy(struct drm_connector *connector)
 {
+	drm_sysfs_connector_remove(connector);
 	drm_connector_cleanup(connector);
 	kfree(connector);
 }
@@ -569,6 +570,7 @@ static struct drm_connector *cirrus_vga_init(struct drm_device *dev)
 			   &cirrus_vga_connector_funcs, DRM_MODE_CONNECTOR_VGA);
 
 	drm_connector_helper_add(connector, &cirrus_vga_connector_helper_funcs);
+	drm_sysfs_connector_add(connector);
 
 	return connector;
 }
