From: Egbert Eich <eich@suse.de>
Date: Sat May 3 19:07:07 2014 +0200
Subject: drm/mgag200: Remove connector in sysfs when unloading the driver
Patch-Mainline: to be upstreamed
References: bnc#876238

Signed-off-by: Egbert Eich <eich@suse.de>
---
 drivers/gpu/drm/mgag200/mgag200_mode.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/gpu/drm/mgag200/mgag200_mode.c
+++ b/drivers/gpu/drm/mgag200/mgag200_mode.c
@@ -1677,6 +1677,7 @@ static void mga_connector_destroy(struct
 {
 	struct mga_connector *mga_connector = to_mga_connector(connector);
 	mgag200_i2c_destroy(mga_connector->i2c);
+	drm_sysfs_connector_remove(connector);
 	drm_connector_cleanup(connector);
 	kfree(connector);
 }
