From 1f76da51f58eda3a027e71e98404b80fcad2d80d Mon Sep 17 00:00:00 2001
From: Roopa Prabhu <roprabhu@cisco.com>
Date: Tue, 10 Aug 2010 18:55:26 +0000
Subject: [PATCH] enic: Mark intr_timer in vnic_enet_config as deprecated
References: bnc#6382582
Git-commit: 1f76da51f58eda3a027e71e98404b80fcad2d80d
Patch-mainline: v2.6.37-rc1~147^2~820

Signed-off-by: Scott Feldman <scofeldm@cisco.com>
Signed-off-by: Roopa Prabhu <roprabhu@cisco.com>
Signed-off-by: Vasanthy Kolluri <vkolluri@cisco.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Brandon Philips <bphilips@suse.de>

---
 drivers/net/enic/vnic_enet.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/net/enic/vnic_enet.h b/drivers/net/enic/vnic_enet.h
index 3b32912..e8740e3 100644
--- a/drivers/net/enic/vnic_enet.h
+++ b/drivers/net/enic/vnic_enet.h
@@ -30,7 +30,7 @@ struct vnic_enet_config {
 	u32 wq_desc_count;
 	u32 rq_desc_count;
 	u16 mtu;
-	u16 intr_timer;
+	u16 intr_timer_deprecated;
 	u8 intr_timer_type;
 	u8 intr_mode;
 	char devname[16];
-- 
1.7.3.1

