From: Gairy Grannum <Gairy.Grannum@Emulex.Com>
Date: Mon, 25 Mar 2013 13:35:55 +0100
Subject: lpfc: Update lpfc version for 8.3.7.10.1p driver release
References: bnc#810882
Patch-Mainline: Submitted to linux-scsi

Signed-off-by: James Smart <james.smart@emulex.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/scsi/lpfc/lpfc_version.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/scsi/lpfc/lpfc_version.h b/drivers/scsi/lpfc/lpfc_version.h
index c41e350..46f351d 100644
--- a/drivers/scsi/lpfc/lpfc_version.h
+++ b/drivers/scsi/lpfc/lpfc_version.h
@@ -18,7 +18,7 @@
  * included with this package.                                     *
  *******************************************************************/
 
-#define LPFC_DRIVER_VERSION "8.3.7.7.1p"
+#define LPFC_DRIVER_VERSION "8.3.7.10.1p"
 #define LPFC_DRIVER_NAME		"lpfc"
 
 /* Used for SLI 2/3 */
-- 
1.7.10.4

