From: Sajin George <sajin.george@emulex.com>
Date: Thu, 16 Jan 2014 09:03:22 +0100
Subject: Update lpfc version for 8.3.7.10.7p driver release
References: bnc#818064
Patch-Mainline: n/a

Signed-off-by: Sajin George <sajin.george@emulex.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/scsi/lpfc/lpfc_version.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/scsi/lpfc/lpfc_version.h b/drivers/scsi/lpfc/lpfc_version.h
index 879d4c4..c374ef9 100644
--- a/drivers/scsi/lpfc/lpfc_version.h
+++ b/drivers/scsi/lpfc/lpfc_version.h
@@ -18,7 +18,7 @@
  * included with this package.                                     *
  *******************************************************************/
 
-#define LPFC_DRIVER_VERSION "8.3.7.10.6p"
+#define LPFC_DRIVER_VERSION "8.3.7.10.7p"
 #define LPFC_DRIVER_NAME		"lpfc"
 
 /* Used for SLI 2/3 */
-- 
1.7.12.4

