From: Gairy Grannum <Gairy.Grannum@Emulex.Com>
Date: Wed, 24 Apr 2013 08:23:38 +0200
Subject: lpfc: Update lpfc version for 8.3.7.10.3p driver release
Patch-Mainline: n/a

Signed-off-by: Gairy Grannum <Gairy.Grannum@emulex.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/scsi/lpfc/lpfc_version.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/scsi/lpfc/lpfc_version.h b/drivers/scsi/lpfc/lpfc_version.h
index b10af59..5025c8d 100644
--- a/drivers/scsi/lpfc/lpfc_version.h
+++ b/drivers/scsi/lpfc/lpfc_version.h
@@ -18,7 +18,7 @@
  * included with this package.                                     *
  *******************************************************************/
 
-#define LPFC_DRIVER_VERSION "8.3.7.10.2p"
+#define LPFC_DRIVER_VERSION "8.3.7.10.3p"
 #define LPFC_DRIVER_NAME		"lpfc"
 
 /* Used for SLI 2/3 */
-- 
1.7.12.4

