From: Gairy Grannum <gairy.grannum@emulex.com>
Date: Thu, 30 May 2013 08:58:56 +0200
Subject: Update lpfc version for 8.3.7.10.6p driver release
Patch-Mainline: n/a
References: bnc#816065

Signed-off-by: Gairy Grannum <Gairy.Grannum@Emulex.Com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/scsi/lpfc/lpfc_version.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/scsi/lpfc/lpfc_version.h b/drivers/scsi/lpfc/lpfc_version.h
index 117abfe..879d4c4 100644
--- a/drivers/scsi/lpfc/lpfc_version.h
+++ b/drivers/scsi/lpfc/lpfc_version.h
@@ -18,7 +18,7 @@
  * included with this package.                                     *
  *******************************************************************/
 
-#define LPFC_DRIVER_VERSION "8.3.7.10.5p"
+#define LPFC_DRIVER_VERSION "8.3.7.10.6p"
 #define LPFC_DRIVER_NAME		"lpfc"
 
 /* Used for SLI 2/3 */
-- 
1.7.12.4

