From: James Smart <james.smart@emulex.com>
Date: Tue, 9 Apr 2013 22:20:28 +0000
Subject: lpfc: Fixed async FCF modified event to in-use FCF failure to trigger recovery
Git-commit: 25aee4070aaac9937a2b88eca3a4333552755130
Patch-Mainline: v3.9

Signed-off-by: James Smart <james.smart@emulex.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/scsi/lpfc/lpfc_init.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/scsi/lpfc/lpfc_init.c b/drivers/scsi/lpfc/lpfc_init.c
index 104e007..a73cb09 100644
--- a/drivers/scsi/lpfc/lpfc_init.c
+++ b/drivers/scsi/lpfc/lpfc_init.c
@@ -4108,9 +4108,9 @@ lpfc_sli4_async_fip_evt(struct lpfc_hba *phba,
 
 		/* If FCF has been in discovered state, perform rediscovery
 		 * only if the FCF with the same index of the in-use FCF got
-		 * modified. Otherwise, do nothing.
+		 * modified during normal operation. Otherwise, do nothing.
 		 */
-		if (phba->fcf.fcf_flag & (FCF_SCAN_DONE | FCF_IN_USE)) {
+		if (phba->pport->port_state > LPFC_FLOGI) {
 			spin_unlock_irq(&phba->hbalock);
 			if (phba->fcf.current_rec.fcf_indx ==
 			    acqe_fip->index) {
-- 
1.7.12.4

