From: David Decotigny <decot@googlers.com>
Date: Wed, 24 Feb 2016 10:58:09 -0800
Subject: net: bridge: use __ethtool_get_ksettings
Patch-mainline: v4.6-rc1
Git-commit: 702b26a24d3d012b9286efb3efbf2b81b23271f5
References: bsc#988035

Signed-off-by: David Decotigny <decot@googlers.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Benjamin Poirier <bpoirier@suse.com>
---
 net/bridge/br_if.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/net/bridge/br_if.c
+++ b/net/bridge/br_if.c
@@ -36,10 +36,10 @@
  */
 static int port_cost(struct net_device *dev)
 {
-	struct ethtool_cmd ecmd;
+	struct ethtool_link_ksettings ecmd;
 
-	if (!__ethtool_get_settings(dev, &ecmd)) {
-		switch (ethtool_cmd_speed(&ecmd)) {
+	if (!__ethtool_get_link_ksettings(dev, &ecmd)) {
+		switch (ecmd.base.speed) {
 		case SPEED_10000:
 			return 2;
 		case SPEED_1000:
