From: Chad Dupuis <chad.dupuis@qlogic.com>
Date: Wed, 18 Sep 2013 07:02:40 -0400
Subject: qla2xxx: Add module parameter to override the default request queue size.
Patch-Mainline: Not yet
References: bnc#826756

Signed-off-by: Petr Uzel <petr.uzel@suse.cz>
Acked-by: Hannes Reinecke <hare@suse.de>

diff --git a/drivers/scsi/qla2xxx/qla_dbg.c b/drivers/scsi/qla2xxx/qla_dbg.c
index 3538c000..4303ff6 100644
--- a/drivers/scsi/qla2xxx/qla_dbg.c
+++ b/drivers/scsi/qla2xxx/qla_dbg.c
@@ -11,7 +11,7 @@
  * ----------------------------------------------------------------------
  * |             Level            |   Last Value Used  |     Holes	|
  * ----------------------------------------------------------------------
- * | Module Init and Probe        |       0x0122       | 0x0117-0x0121  |
+ * | Module Init and Probe        |       0x0128       | 0x0117-0x0121  |
  * | Mailbox commands             |       0x1129       |		|
  * | Device Discovery             |       0x2083       |		|
  * | Queue Command and IO tracing |       0x3031       |     0x3008     |
diff --git a/drivers/scsi/qla2xxx/qla_def.h b/drivers/scsi/qla2xxx/qla_def.h
index 7810074..b37e370 100644
--- a/drivers/scsi/qla2xxx/qla_def.h
+++ b/drivers/scsi/qla2xxx/qla_def.h
@@ -182,6 +182,7 @@
 #define REQUEST_ENTRY_CNT_2100		128	/* Number of request entries. */
 #define REQUEST_ENTRY_CNT_2200		2048	/* Number of request entries. */
 #define REQUEST_ENTRY_CNT_24XX		2048	/* Number of request entries. */
+#define REQUEST_ENTRY_CNT_MAX		16384	/* Max # of request entries */
 #define RESPONSE_ENTRY_CNT_2100		64	/* Number of response entries.*/
 #define RESPONSE_ENTRY_CNT_2300		512	/* Number of response entries.*/
 #define RESPONSE_ENTRY_CNT_MQ		128	/* Number of response entries.*/
diff --git a/drivers/scsi/qla2xxx/qla_os.c b/drivers/scsi/qla2xxx/qla_os.c
index dc7f3b3..19a0e33 100644
--- a/drivers/scsi/qla2xxx/qla_os.c
+++ b/drivers/scsi/qla2xxx/qla_os.c
@@ -13,6 +13,7 @@
 #include <linux/mutex.h>
 #include <linux/kobject.h>
 #include <linux/slab.h>
+#include <linux/log2.h>
 
 #include <scsi/scsi_tcq.h>
 #include <scsi/scsicam.h>
@@ -214,6 +215,12 @@ MODULE_PARM_DESC(ql2xasynclogin,
 		"Default is 1 - Use asynchronous login if possible.");
 
 
+int ql2xreq_que_size = 0;
+module_param(ql2xreq_que_size, int, S_IRUGO);
+MODULE_PARM_DESC(ql2xreq_que_size,
+		"Override the default request queue size. Valid values must be "
+		"a power of 2 and between 2048 and 16384 inclusive.");
+
 /*
  * SCSI host template entry points
  */
@@ -2291,6 +2298,25 @@ qla2x00_probe_one(struct pci_dev *pdev, const struct pci_device_id *id)
 		ha->nvram_conf_off = FARX_ACCESS_NVRAM_CONF;
 		ha->nvram_data_off = FARX_ACCESS_NVRAM_DATA;
 	}
+
+	/* Override the default request queue size if requested. */
+	if (ql2xreq_que_size >= req_length &&
+	    ql2xreq_que_size <= REQUEST_ENTRY_CNT_MAX &&
+	    is_power_of_2(ql2xreq_que_size) &&
+	    IS_FWI2_CAPABLE(ha) && !IS_QLA82XX(ha))
+		req_length = ql2xreq_que_size;
+	else if (ql2xreq_que_size > 0) {
+		if (!IS_FWI2_CAPABLE(ha) || IS_QLA82XX(ha))
+			ql_log_pci(ql_log_warn, pdev, 0x0127,
+			    "Cannot change the request queue size for this "
+			    "adapter.\n");
+		else
+			ql_log_pci(ql_log_warn, pdev, 0x128,
+			    "Changes in request queue size must be a power of "
+			    "2 between %d and %d, inclusive.\n", req_length,
+			    REQUEST_ENTRY_CNT_MAX);
+	}
+
 	ql_dbg_pci(ql_dbg_init, pdev, 0x001e,
 	    "mbx_count=%d, req_length=%d, "
 	    "rsp_length=%d, max_loop_id=%d, init_cb_size=%d, "
