From: "Amrani, Ram" <Ram.Amrani@cavium.com>
Date: Tue, 24 Jan 2017 12:01:31 +0200
Subject: RDMA/qedr: Add uapi header qedr-abi.h
Patch-mainline: v4.10-rc6
Git-commit: 20f5e10ef8bcf29a915642245b66e5a132e38fc4
References: bsc#1022604

Signed-off-by: Ram Amrani <Ram.Amrani@cavium.com>
Signed-off-by: Doug Ledford <dledford@redhat.com>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 include/uapi/rdma/Kbuild | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/uapi/rdma/Kbuild b/include/uapi/rdma/Kbuild
index f2ca00ba27d4..029849ffaf6a 100644
--- a/include/uapi/rdma/Kbuild
+++ b/include/uapi/rdma/Kbuild
@@ -16,3 +16,4 @@ header-y += mthca-abi.h
 header-y += nes-abi.h
 header-y += ocrdma-abi.h
 header-y += hns-abi.h
+header-y += qedr-abi.h
-- 
1.8.5.6

