From: Joseph Mann <jmann@emulex.com>
Date: Thu, 17 Jan 2013 09:29:29 +0100
Subject: Update lpfc version for 8.3.7.7.1p driver release
References: FATE#313818, bnc#777561
Patch-Mainline: n/a

Signed-off-by: Joseph Mann <jmann@emulex.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/scsi/lpfc/lpfc_version.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/scsi/lpfc/lpfc_version.h b/drivers/scsi/lpfc/lpfc_version.h
index 515e761..c41e350 100644
--- a/drivers/scsi/lpfc/lpfc_version.h
+++ b/drivers/scsi/lpfc/lpfc_version.h
@@ -18,7 +18,7 @@
  * included with this package.                                     *
  *******************************************************************/
 
-#define LPFC_DRIVER_VERSION "8.3.5.48.2p"
+#define LPFC_DRIVER_VERSION "8.3.7.7.1p"
 #define LPFC_DRIVER_NAME		"lpfc"
 
 /* Used for SLI 2/3 */
-- 
1.7.4.2

