From a90131763dfd9a95f43071f0b67a737e6ca6a6c0 Mon Sep 17 00:00:00 2001
From: Johan Hovold <jhovold@gmail.com>
Date: Sat, 15 May 2010 17:53:50 +0200
Subject: USB: cp210x: use generic open to start reading from device
Patch-mainline: v2.6.35
Git-commit: a90131763dfd9a95f43071f0b67a737e6ca6a6c0

Signed-off-by: Johan Hovold <jhovold@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/cp210x.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/serial/cp210x.c
+++ b/drivers/usb/serial/cp210x.c
@@ -401,7 +401,7 @@ static int cp210x_open(struct tty_struct
 		return -EPROTO;
 	}
 
-	result = usb_serial_generic_submit_read_urb(port, GFP_KERNEL);
+	result = usb_serial_generic_open(tty, port);
 	if (result)
 		return result;
 
