From 4e0961d53020b0fed83168d469ce44e1b3f25898 Mon Sep 17 00:00:00 2001
From: Ben Hutchings <ben@decadent.org.uk>
Date: Wed, 13 Jan 2010 23:39:43 +0000
Subject: USB: isight-firmware: declare MODULE_FIRMWARE
Patch-mainline: v2.6.34
Git-commit: 4e0961d53020b0fed83168d469ce44e1b3f25898

Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/misc/isight_firmware.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/usb/misc/isight_firmware.c
+++ b/drivers/usb/misc/isight_firmware.c
@@ -112,6 +112,8 @@ out:
 	return ret;
 }
 
+MODULE_FIRMWARE("isight.fw");
+
 static void isight_firmware_disconnect(struct usb_interface *intf)
 {
 }
