From 55c6d85cabb33461139e430f48b9dea794d46f67 Mon Sep 17 00:00:00 2001
From: Johan Hovold <jhovold@gmail.com>
Date: Thu, 13 May 2010 21:33:09 +0200
Subject: USB: ipw: remove dummy port_probe and port_remove
Patch-mainline: v2.6.35
Git-commit: 55c6d85cabb33461139e430f48b9dea794d46f67

Signed-off-by: Johan Hovold <jhovold@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/ipw.c |   13 -------------
 1 file changed, 13 deletions(-)

--- a/drivers/usb/serial/ipw.c
+++ b/drivers/usb/serial/ipw.c
@@ -405,17 +405,6 @@ static int ipw_write(struct tty_struct *
 	return count;
 }
 
-static int ipw_probe(struct usb_serial_port *port)
-{
-	return 0;
-}
-
-static int ipw_disconnect(struct usb_serial_port *port)
-{
-	usb_set_serial_port_data(port, NULL);
-	return 0;
-}
-
 static struct usb_serial_driver ipw_device = {
 	.driver = {
 		.owner =	THIS_MODULE,
@@ -428,8 +417,6 @@ static struct usb_serial_driver ipw_devi
 	.open =			ipw_open,
 	.close =		ipw_close,
 	.dtr_rts =		ipw_dtr_rts,
-	.port_probe = 		ipw_probe,
-	.port_remove =		ipw_disconnect,
 	.write =		ipw_write,
 	.write_bulk_callback =	ipw_write_bulk_callback,
 	.read_bulk_callback =	ipw_read_bulk_callback,
