From: Igor Pylypiv <ipylypiv@silver-peak.com>
Date: Sat, 17 Mar 2018 18:17:58 -0700
Subject: vmxnet3: remove unused flag "rxcsum" from struct vmxnet3_adapter
Patch-mainline: v4.16-rc7
Git-commit: 8137a8e2190975ed9060111ce13000792360aba3
References: bsc#1091860 bsc#1098253

Signed-off-by: Igor Pylypiv <ipylypiv@silver-peak.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/net/vmxnet3/vmxnet3_int.h |    2 --
 1 file changed, 2 deletions(-)

--- a/drivers/net/vmxnet3/vmxnet3_int.h
+++ b/drivers/net/vmxnet3/vmxnet3_int.h
@@ -343,8 +343,6 @@ struct vmxnet3_adapter {
 	u8			__iomem *hw_addr1; /* for BAR 1 */
 	u8                              version;
 
-	bool				rxcsum;
-
 #ifdef VMXNET3_RSS
 	struct UPT1_RSSConf		*rss_conf;
 	bool				rss;
