Subject: Introduce a private switchback method to preserve kABI
From: Oliver Neukum <oneukum@suse.de>
References: bnc#772427
Patch-Mainline: Never

Do this in a manner compatible to the kABI

Signed-off-by: Oliver Neukum <oneukum@suse.de>

--- linux-3.0-SLE11-SP2/drivers/usb/host/xhci.h.alt	2012-08-10 12:38:06.000000000 +0200
+++ linux-3.0-SLE11-SP2/drivers/usb/host/xhci.h	2012-08-10 12:39:32.000000000 +0200
@@ -1468,6 +1468,7 @@ struct xhci_hcd {
 #define	XHCI_SW_BW_CHECKING	(1 << 8)
 #define XHCI_AMD_0x96_HOST	(1 << 9)
 #define XHCI_TRUST_TX_LENGTH	(1 << 10)
+#define XHCI_SPURIOUS_REBOOT	(1 << 13)
 	unsigned int		num_active_eps;
 	unsigned int		limit_active_eps;
 	/* There are two roothubs to keep track of bus suspend info for */
--- linux-3.0-SLE11-SP2/drivers/usb/host/xhci.c.alt	2012-08-10 12:34:30.000000000 +0200
+++ linux-3.0-SLE11-SP2/drivers/usb/host/xhci.c	2012-08-10 12:37:24.000000000 +0200
@@ -634,6 +634,11 @@ void xhci_stop(struct usb_hcd *hcd)
 		    xhci_readl(xhci, &xhci->op_regs->status));
 }
 
+static void usb_disable_xhci_ports(struct pci_dev *xhci_pdev)
+{
+	pci_write_config_dword(xhci_pdev, USB_INTEL_USB3_PSSEN, 0x0);
+	pci_write_config_dword(xhci_pdev, USB_INTEL_XUSB2PR, 0x0);
+}
 /*
  * Shutdown HC (not bus-specific)
  *
