From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Wed, 30 May 2018 22:11:35 +0200
Subject: [PATCH] Linux 4.4.135
References: bnc#1012382
Patch-mainline: 4.4.135
Git-commit: 50eb02ed89920f753202d703541bebbd9d8c3dd8

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 119dbcb4f311..9b14a5a224c5 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 4
-SUBLEVEL = 134
+SUBLEVEL = 135
 EXTRAVERSION =
 NAME = Blurry Fish Butt
 
-- 
2.17.1

