From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Wed, 2 May 2018 07:53:43 -0700
Subject: [PATCH] Linux 4.4.131
References: bnc#1012382
Patch-mainline: 4.4.131
Git-commit: 8719027e78c77c31d338c616265caeb25887cf84

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 151477d4d5e5..6ec65396a56d 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 4
-SUBLEVEL = 130
+SUBLEVEL = 131
 EXTRAVERSION =
 NAME = Blurry Fish Butt
 
-- 
2.16.3

