From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Sat, 24 Mar 2018 10:58:49 +0100
Subject: [PATCH] Linux 4.4.124
References: bnc#1012382
Patch-mainline: 4.4.124
Git-commit: b766b14a48de577e5060e4d8762f1e4a34229d2b

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index cbcc04da790a..bbaf3fd0d1ef 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 4
-SUBLEVEL = 123
+SUBLEVEL = 124
 EXTRAVERSION =
 NAME = Blurry Fish Butt
 
-- 
2.16.3

