From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Sun, 22 Jul 2018 14:25:54 +0200
Subject: [PATCH] Linux 4.4.143
References: bnc#1012382
Patch-mainline: 4.4.143
Git-commit: a8ea6276d00555387deaaa5eaeb380cd5c17bdc9

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 75d6176c8786..54690fee0485 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 4
-SUBLEVEL = 142
+SUBLEVEL = 143
 EXTRAVERSION =
 NAME = Blurry Fish Butt
 
-- 
2.18.0

