From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Fri, 27 Oct 2017 10:23:18 +0200
Subject: [PATCH] Linux 4.4.95
References: bnc#1012382
Patch-mainline: 4.4.95
Git-commit: 9b36699635c54b2e56ec3fc07a750dc465542a6d

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index ff9d6bbf2210..57e1ea2a189a 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 4
-SUBLEVEL = 94
+SUBLEVEL = 95
 EXTRAVERSION =
 NAME = Blurry Fish Butt
 
-- 
2.14.3

