From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Thu, 19 Jul 2018 15:35:58 +0200
Subject: [PATCH] Linux 4.4.142
References: bnc#1012382
Patch-mainline: 4.4.142
Git-commit: ecb99897516f0bb433f3e79df3b1958c80ac4810

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 3fc39e41dbde..75d6176c8786 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 4
-SUBLEVEL = 141
+SUBLEVEL = 142
 EXTRAVERSION =
 NAME = Blurry Fish Butt
 
-- 
2.18.0

