From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Sat, 18 Nov 2017 11:11:07 +0100
Subject: [PATCH] Linux 4.4.99
References: bnc#1012382
Patch-mainline: 4.4.99
Git-commit: 0cbac004e67307949714c176c8a7af9c1da980b9

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 5d62e23347f9..0b5d9e20eee2 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 4
-SUBLEVEL = 98
+SUBLEVEL = 99
 EXTRAVERSION =
 NAME = Blurry Fish Butt
 
-- 
2.15.0

