From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Sun, 18 Mar 2018 11:17:54 +0100
Subject: [PATCH] Linux 4.4.122
References: bnc#1012382
Patch-mainline: 4.4.122
Git-commit: b8ea1f9b32599d3cc0ce81636d3842aee2ce302b

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 0e22e4eb85cc..3eb21d269b42 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 4
-SUBLEVEL = 121
+SUBLEVEL = 122
 EXTRAVERSION =
 NAME = Blurry Fish Butt
 
-- 
2.16.2

