From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Wed, 13 Jun 2018 16:15:29 +0200
Subject: [PATCH] Linux 4.4.137
References: bnc#1012382
Patch-mainline: 4.4.137
Git-commit: ed90fd0cfe4ee5faa42a39605f430f13f39556da

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index a05a7a005715..44efd1252ab8 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 4
-SUBLEVEL = 136
+SUBLEVEL = 137
 EXTRAVERSION =
 NAME = Blurry Fish Butt
 
-- 
2.17.1

