From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Wed, 15 Aug 2018 17:42:11 +0200
Subject: [PATCH] Linux 4.4.148
References: bnc#1012382
Patch-mainline: 4.4.148
Git-commit: 30a97c1e2dc39f45d9deeeccc2733278fc285d5e

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index ee92a12e3a4b..9b795164122e 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 4
-SUBLEVEL = 147
+SUBLEVEL = 148
 EXTRAVERSION =
 NAME = Blurry Fish Butt
 
-- 
2.18.0

