From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Sun, 8 Apr 2018 11:52:02 +0200
Subject: [PATCH] Linux 4.4.127
References: bnc#1012382
Patch-mainline: 4.4.127
Git-commit: 2cad7a1d1316d2955982c50b3b9a6809dff5a7d3

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 2b699c5f6de4..58c0390573c1 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 4
-SUBLEVEL = 126
+SUBLEVEL = 127
 EXTRAVERSION =
 NAME = Blurry Fish Butt
 
-- 
2.16.3

