From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Thu, 22 Mar 2018 09:23:32 +0100
Subject: [PATCH] Linux 4.4.123
References: bnc#1012382
Patch-mainline: 4.4.123
Git-commit: 3753696b0aee3f3f55e161ef41d2cb0ea7e6e225

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 3eb21d269b42..cbcc04da790a 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 4
-SUBLEVEL = 122
+SUBLEVEL = 123
 EXTRAVERSION =
 NAME = Blurry Fish Butt
 
-- 
2.16.2

