From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Sat, 18 Aug 2018 10:45:38 +0200
Subject: [PATCH] Linux 4.4.150
References: bnc#1012382
Patch-mainline: 4.4.150
Git-commit: 7dc18ebc3101229d5238a2dc740804cd4836b383

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index e7c46ece5f27..7789195c6a59 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 4
-SUBLEVEL = 149
+SUBLEVEL = 150
 EXTRAVERSION =
 NAME = Blurry Fish Butt
 
-- 
2.18.0

