From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Fri, 17 Aug 2018 20:56:45 +0200
Subject: [PATCH] Linux 4.4.149
References: bnc#1012382
Patch-mainline: 4.4.149
Git-commit: 45cf1802a1057650768430cf3168ff7a02163338

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index cacb45c3173a..e7c46ece5f27 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 4
-SUBLEVEL = 148
+SUBLEVEL = 149
 EXTRAVERSION =
 NAME = Blurry Fish Butt
 
-- 
2.18.0

