From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Tue, 3 Jul 2018 11:21:35 +0200
Subject: [PATCH] Linux 4.4.139
References: bnc#1012382
Patch-mainline: 4.4.139
Git-commit: 16af098616e73b39c4d98b0d4358cc29a9539c98

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 1a8c0fc6b997..20a11fd36656 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 4
-SUBLEVEL = 138
+SUBLEVEL = 139
 EXTRAVERSION =
 NAME = Blurry Fish Butt
 
-- 
2.18.0

