From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Sat, 31 Mar 2018 18:12:35 +0200
Subject: [PATCH] Linux 4.4.126
References: bnc#1012382
Patch-mainline: 4.4.126
Git-commit: 8ff8cb8ec27effc5a0b04fee2c383b231a19f691

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 7dcafa5dcd34..2b699c5f6de4 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 4
-SUBLEVEL = 125
+SUBLEVEL = 126
 EXTRAVERSION =
 NAME = Blurry Fish Butt
 
-- 
2.16.3

