From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Wed, 16 May 2018 10:06:53 +0200
Subject: [PATCH] Linux 4.4.132
References: bnc#1012382
Patch-mainline: 4.4.132
Git-commit: 69847b97f21141c2d4f6e9c1b22d932b7add3182

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 6ec65396a56d..ace4a655548a 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 4
-SUBLEVEL = 131
+SUBLEVEL = 132
 EXTRAVERSION =
 NAME = Blurry Fish Butt
 
-- 
2.16.3

