From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Fri, 13 Apr 2018 19:50:28 +0200
Subject: [PATCH] Linux 4.4.128
References: bnc#1012382
Patch-mainline: 4.4.128
Git-commit: dbb78762365d0e506110f50dd5a0b95d11e46bf8

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 58c0390573c1..575459bb47eb 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 4
-SUBLEVEL = 127
+SUBLEVEL = 128
 EXTRAVERSION =
 NAME = Blurry Fish Butt
 
-- 
2.16.3

