From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Tue, 24 Apr 2018 09:32:12 +0200
Subject: [PATCH] Linux 4.4.129
References: bnc#1012382
Patch-mainline: 4.4.129
Git-commit: 8e2def054b2b088d18d7009aecf470aa62ab360e

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 575459bb47eb..096d7e867b6c 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 4
-SUBLEVEL = 128
+SUBLEVEL = 129
 EXTRAVERSION =
 NAME = Blurry Fish Butt
 
-- 
2.16.3

