From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Wed, 6 Jun 2018 16:46:24 +0200
Subject: [PATCH] Linux 4.4.136
References: bnc#1012382
Patch-mainline: 4.4.136
Git-commit: dc45cafe612ec6960fe728f3260a0b751c73f4aa

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 9b14a5a224c5..a05a7a005715 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 4
-SUBLEVEL = 135
+SUBLEVEL = 136
 EXTRAVERSION =
 NAME = Blurry Fish Butt
 
-- 
2.17.1

