From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Sun, 11 Mar 2018 16:19:47 +0100
Subject: [PATCH] Linux 4.4.121
References: bnc#1012382
Patch-mainline: 4.4.121
Git-commit: 8b5ab55d254f36e89b1b53aeac7223d2d102483e

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index d874d7de0cde..0e22e4eb85cc 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 4
-SUBLEVEL = 120
+SUBLEVEL = 121
 EXTRAVERSION =
 NAME = Blurry Fish Butt
 
-- 
2.16.2

