From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Sun, 29 Apr 2018 07:50:07 +0200
Subject: [PATCH] Linux 4.4.130
References: bnc#1012382
Patch-mainline: 4.4.130
Git-commit: 34a220d573fcb5d66ad466611eb80f65b6af98d4

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 096d7e867b6c..151477d4d5e5 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 4
-SUBLEVEL = 129
+SUBLEVEL = 130
 EXTRAVERSION =
 NAME = Blurry Fish Butt
 
-- 
2.16.3

