From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Mon, 6 Aug 2018 16:24:42 +0200
Subject: [PATCH] Linux 4.4.146
References: bnc#1012382
Patch-mainline: 4.4.146
Git-commit: bffa1e42b3713aa7911cc3f9a6e5a2dbbf1dc789

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index be31491a2d67..030f5af05f4e 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 4
-SUBLEVEL = 145
+SUBLEVEL = 146
 EXTRAVERSION =
 NAME = Blurry Fish Butt
 
-- 
2.18.0

