From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Wed, 30 May 2018 07:49:18 +0200
Subject: [PATCH] Linux 4.4.134
References: bnc#1012382
Patch-mainline: 4.4.134
Git-commit: 712a231848027a1728d90784cc1f7aec6c017abd

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index ac52ee65685b..119dbcb4f311 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 4
-SUBLEVEL = 133
+SUBLEVEL = 134
 EXTRAVERSION =
 NAME = Blurry Fish Butt
 
-- 
2.17.1

