From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Thu, 9 Aug 2018 12:19:28 +0200
Subject: [PATCH] Linux 4.4.147
References: bnc#1012382
Patch-mainline: 4.4.147
Git-commit: 8404ae6c8c9ff23a06cf38112e83002e1088bfe1

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 030f5af05f4e..ee92a12e3a4b 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 4
-SUBLEVEL = 146
+SUBLEVEL = 147
 EXTRAVERSION =
 NAME = Blurry Fish Butt
 
-- 
2.18.0

