From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Date: Wed, 28 Mar 2018 18:40:17 +0200
Subject: [PATCH] Linux 4.4.125
References: bnc#1012382
Patch-mainline: 4.4.125
Git-commit: aec8e72ebde913a205f5c466d718bd5fe14ab440

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 240be8971e89..7dcafa5dcd34 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 VERSION = 4
 PATCHLEVEL = 4
-SUBLEVEL = 124
+SUBLEVEL = 125
 EXTRAVERSION =
 NAME = Blurry Fish Butt
 
-- 
2.16.3

