From: Yinghai Lu <yinghai.lu@oracle.com>
Subject: workaround hard-wired bus number
References: bnc#735909
Patch-Mainline: not yet

Signed-off-by: Thomas Renninger <trenn@suse.de>

---
 drivers/pci/probe.c |    6 ++++++
 1 file changed, 6 insertions(+)

Index: linux-3.0-SLE11-SP2/drivers/pci/probe.c
===================================================================
--- linux-3.0-SLE11-SP2.orig/drivers/pci/probe.c
+++ linux-3.0-SLE11-SP2/drivers/pci/probe.c
@@ -657,6 +657,12 @@ int __devinit pci_scan_bridge(struct pci
 	dev_dbg(&dev->dev, "scanning [bus %02x-%02x] behind bridge, pass %d\n",
 		secondary, subordinate, pass);
 
+	/* some bridge primary bus is hard wired to 0 */
+	if (!primary && (primary != bus->number) && secondary && subordinate) {
+		dev_warn(&dev->dev, "Primary bus is hard wired to 0\n");
+		primary = bus->number;
+	}
+
 	/* Check if setup is sensible at all */
 	if (!pass &&
 	    (primary != bus->number || secondary <= bus->number)) {
