From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 4 Mar 2016 14:37:16 -0500
Subject: dm: add missing newline between DM_DEBUG_BLOCK_STACK_TRACING and
References: bnc#974525,FATE#320741
Patch-Mainline: v4.6-rc1
Git-commit: 3f0680402c2d0da58a8b06f7e55d387591b55e61
 DM_BUFIO

Signed-off-by: Mike Snitzer <snitzer@redhat.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/md/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/md/Kconfig b/drivers/md/Kconfig
index 2ae4091..02a5345 100644
--- a/drivers/md/Kconfig
+++ b/drivers/md/Kconfig
@@ -249,6 +249,7 @@ config DM_DEBUG_BLOCK_STACK_TRACING
 	 block manager locking used by thin provisioning and caching.
 
 	 If unsure, say N.
+
 config DM_BIO_PRISON
        tristate
        depends on BLK_DEV_DM
-- 
1.8.5.6

