From: Anup Limbu <anuplimbu14@gmail.com>
Date: Wed, 25 Nov 2015 15:46:05 +0530
Subject: dm bufio: use BUG_ON instead of conditional call to BUG
References: bnc#974525,FATE#320741
Patch-Mainline: v4.6-rc1
Git-commit: 86a49e2dac3064557d45850a9c63f7cc1e6fe041

Signed-off-by: Anup Limbu <anuplimbu14@gmail.com>
Signed-off-by: Mike Snitzer <snitzer@redhat.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/md/dm-bufio.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/md/dm-bufio.c b/drivers/md/dm-bufio.c
index bccc7e6..6b832e0 100644
--- a/drivers/md/dm-bufio.c
+++ b/drivers/md/dm-bufio.c
@@ -1928,8 +1928,7 @@ static void __exit dm_bufio_exit(void)
 		bug = 1;
 	}
 
-	if (bug)
-		BUG();
+	BUG_ON(bug);
 }
 
 module_init(dm_bufio_init)
-- 
1.8.5.6

