From: Mike Snitzer <snitzer@redhat.com>
Date: Sat, 30 Jan 2016 21:10:03 -0500
Subject: dm: remove unused dm_get_rq_mapinfo()
References: bnc#974525,FATE#320741
Patch-Mainline: v4.6-rc1
Git-commit: ae6ad75e5c3c51f99c57e289058c5ec4c8701963

Signed-off-by: Mike Snitzer <snitzer@redhat.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/md/dm.c | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/drivers/md/dm.c b/drivers/md/dm.c
index db7a51a..30302df 100644
--- a/drivers/md/dm.c
+++ b/drivers/md/dm.c
@@ -106,14 +106,6 @@ struct dm_rq_clone_bio_info {
 	struct bio clone;
 };
 
-union map_info *dm_get_rq_mapinfo(struct request *rq)
-{
-	if (rq && rq->end_io_data)
-		return &((struct dm_rq_target_io *)rq->end_io_data)->info;
-	return NULL;
-}
-EXPORT_SYMBOL_GPL(dm_get_rq_mapinfo);
-
 #define MINOR_ALLOCED ((void *)-1)
 
 /*
-- 
1.8.5.6

