From: Wen Xiong <wenxiong@us.ibm.com>
Subject: Improve safety of dlpar remove when rdma traffic is active
References: bnc#590727
Patch-Mainline: v2.6.34-rc2
Git-commit: 69960a275efc9d82797bbbe2460a2d6c9cace314

Signed-off-by: Thomas Renninger <trenn@suse.de>

Index: linux-2.6.32-637639/drivers/infiniband/hw/cxgb3/iwch.c
===================================================================
--- linux-2.6.32-637639.orig/drivers/infiniband/hw/cxgb3/iwch.c
+++ linux-2.6.32-637639/drivers/infiniband/hw/cxgb3/iwch.c
@@ -148,6 +148,7 @@ static void close_rnic_dev(struct t3cdev
 	list_for_each_entry_safe(dev, tmp, &dev_list, entry) {
 		if (dev->rdev.t3cdev_p == tdev) {
 			dev->rdev.flags = CXIO_ERROR_FATAL;
+			synchronize_net();
 			list_del(&dev->entry);
 			iwch_unregister_device(dev);
 			cxio_rdev_close(&dev->rdev);
@@ -174,6 +175,7 @@ static void iwch_event_handler(struct t3
 	switch (evt) {
 	case OFFLOAD_STATUS_DOWN: {
 		rdev->flags = CXIO_ERROR_FATAL;
+		synchronize_net();
 		event.event  = IB_EVENT_DEVICE_FATAL;
 		break;
 		}
Index: linux-2.6.32-637639/drivers/net/cxgb3/cxgb3_main.c
===================================================================
--- linux-2.6.32-637639.orig/drivers/net/cxgb3/cxgb3_main.c
+++ linux-2.6.32-637639/drivers/net/cxgb3/cxgb3_main.c
@@ -438,7 +438,7 @@ static void free_irq_resources(struct ad
 static int await_mgmt_replies(struct adapter *adap, unsigned long init_cnt,
 			      unsigned long n)
 {
-	int attempts = 5;
+	int attempts = 20;
 
 	while (adap->sge.qs[0].rspq.offload_pkts < init_cnt + n) {
 		if (!--attempts)


