From: Wen Xiong <wenxiong@us.ibm.com>
Subject: Improve safety of dlpar remove when rdma traffic is active
References: bnc#590727
Patch-Mainline: Not yet

Signed-off-by: Thomas Renninger <trenn@suse.de>

diff -Nuarp rc1.orig/drivers/infiniband/hw/cxgb3/iwch.c rc1.dlpar/drivers/infiniband/hw/cxgb3/iwch.c
--- rc1.orig/drivers/infiniband/hw/cxgb3/iwch.c	2010-03-23 23:16:39.000000000 -0500
+++ rc1.dlpar/drivers/infiniband/hw/cxgb3/iwch.c	2010-03-23 23:19:15.000000000 -0500
@@ -148,6 +148,7 @@ static void close_rnic_dev(struct t3cdev
 	list_for_each_entry_safe(dev, tmp, &dev_list, entry) {
 		if (dev->rdev.t3cdev_p == tdev) {
 			dev->rdev.flags = CXIO_ERROR_FATAL;
+			synchronize_net();
 			list_del(&dev->entry);
 			iwch_unregister_device(dev);
 			cxio_rdev_close(&dev->rdev);
@@ -174,6 +175,7 @@ static void iwch_event_handler(struct t3
 	switch (evt) {
 	case OFFLOAD_STATUS_DOWN: {
 		rdev->flags = CXIO_ERROR_FATAL;
+		synchronize_net();
 		event.event  = IB_EVENT_DEVICE_FATAL;
 		break;
 		}
diff -Nuarp rc1.orig/drivers/net/cxgb3/cxgb3_main.c rc1.dlpar/drivers/net/cxgb3/cxgb3_main.c
--- rc1.orig/drivers/net/cxgb3/cxgb3_main.c	2010-03-23 23:15:58.000000000 -0500
+++ rc1.dlpar/drivers/net/cxgb3/cxgb3_main.c	2010-03-23 23:18:04.000000000 -0500
@@ -438,7 +438,7 @@ static void free_irq_resources(struct ad
 static int await_mgmt_replies(struct adapter *adap, unsigned long init_cnt,
 			      unsigned long n)
 {
-	int attempts = 5;
+	int attempts = 20;
 
 	while (adap->sge.qs[0].rspq.offload_pkts < init_cnt + n) {
 		if (!--attempts)
@@ -1274,6 +1274,7 @@ static void cxgb_down(struct adapter *ad
 
 	free_irq_resources(adapter);
 	quiesce_rx(adapter);
+	t3_sge_stop(adapter);
 	flush_workqueue(cxgb3_wq);	/* wait for external IRQ handler */
 }
 
