From: Thomas Renninger <trenn@suse.de>
Subject: Whitelist HP ProLiant DL980 to use pci=use_crs by default
References: bnc#578572
Patch-Mainline: never - mainline enables use_crs by BIOS date

Signed-off-by: Thomas Renninger <trenn@suse.de>
---
 arch/x86/pci/acpi.c |    8 ++++++++
 1 file changed, 8 insertions(+)

Index: linux-2.6.32-SLE11-SP1/arch/x86/pci/acpi.c
===================================================================
--- linux-2.6.32-SLE11-SP1.orig/arch/x86/pci/acpi.c
+++ linux-2.6.32-SLE11-SP1/arch/x86/pci/acpi.c
@@ -32,6 +32,14 @@ static const struct dmi_system_id pci_us
 			DMI_MATCH(DMI_PRODUCT_NAME, "x3800"),
 		},
 	},
+	{
+		.callback = set_use_crs,
+		.ident = "HP ProLiant DL980 G7",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "Hewlett-Packard"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "ProLiant DL980 G7"),
+		},
+	},
 	{}
 };
 
