From: David Wagner <david.wagner@qlogic.com>
Subject: Skip MSI-X for qla2xxx
References: bnc#693513
Patch-mainline: Never, will be pulled from driver updates

Acked-by: Goldwyn Rodrigues <rgoldwyn@suse.de>

Index: linux-2.6.32-SLES11-SP1/drivers/scsi/qla2xxx/qla_isr.c
===================================================================
--- linux-2.6.32-SLES11-SP1.orig/drivers/scsi/qla2xxx/qla_isr.c	2011-07-27 18:26:32.000000000 -0500
+++ linux-2.6.32-SLES11-SP1/drivers/scsi/qla2xxx/qla_isr.c	2011-08-04 12:03:01.000000000 -0500
@@ -2203,8 +2203,7 @@
 		goto skip_msi;
 	}
 
-	if (IS_QLA2432(ha) && (ha->pdev->revision < QLA_MSIX_CHIP_REV_24XX ||
-		!QLA_MSIX_FW_MODE_1(ha->fw_attributes))) {
+	if (IS_QLA2432(ha) && (ha->pdev->revision < QLA_MSIX_CHIP_REV_24XX)) {
 		DEBUG2(qla_printk(KERN_WARNING, ha,
 		"MSI-X: Unsupported ISP2432 (0x%X, 0x%X).\n",
 			ha->pdev->revision, ha->fw_attributes));
