From: Hannes Reinecke <hare@suse.de>
Date: Wed, 9 Jan 2013 12:36:58 +0100
Subject: [SCSI] Remove duplicate call to blk_cleanup_queue()
References: FATE#313629
Patch-Mainline: n/a

Merge error in earlier patch, found by code inspection.

Signed-off-by: Hannes Reinecke <hare@suse.de>

diff --git a/drivers/scsi/scsi_sysfs.c b/drivers/scsi/scsi_sysfs.c
index aa7c81b..ce5224c 100644
--- a/drivers/scsi/scsi_sysfs.c
+++ b/drivers/scsi/scsi_sysfs.c
@@ -981,8 +981,6 @@ void __scsi_remove_device(struct scsi_device *sdev)
 		sdev->host->hostt->slave_destroy(sdev);
 	transport_destroy_device(dev);
 
-	/* Freeing the queue signals to block that we're done */
-	blk_cleanup_queue(sdev->request_queue);
 	put_device(dev);
 }
 
