From: Alexandra Kossovsky <Alexandra.Kossovsky@oktetlabs.ru>
Subject: tcp: Fix OOB POLLIN avoidance.
References: bnc#598493
Patch-mainline: 2.6.34
X-Git-Commit: b634f87522dff87712df8bda2a6c9061954d552a
Acked-by: Jiri Bohac <jbohac@suse.cz>

Fixes kernel bugzilla #15541

Signed-off-by: David S. Miller <davem@davemloft.net>

diff --git a/net/ipv4/tcp.c b/net/ipv4/tcp.c
index 5901010..ae16f80 100644
--- a/net/ipv4/tcp.c
+++ b/net/ipv4/tcp.c
@@ -429,7 +429,7 @@ unsigned int tcp_poll(struct file *file, struct socket *sock, poll_table *wait)
 		if (tp->urg_seq == tp->copied_seq &&
 		    !sock_flag(sk, SOCK_URGINLINE) &&
 		    tp->urg_data)
-			target--;
+			target++;
 
 		/* Potential race condition. If read of tp below will
 		 * escape above sk->sk_state, we can be illegally awaken
