From: Goldwyn Rodrigues <rgoldwyn@suse.de>
Patch-mainline: 2.6.38-rc2
Git-Commit: eda77982729b7170bdc9e8855f0682edf322d277
Subject: xfs: serialise unaligned direct IOs
References: bnc#707125

Signed-off-by: Goldwyn Rodrigues <rgoldwyn@suse.de>
Reviewed-by: Alex Elder <aelder@sgi.com>

Index: linux-2.6.32-SLES11-SP1/fs/xfs/linux-2.6/xfs_lrw.c
===================================================================
--- linux-2.6.32-SLES11-SP1.orig/fs/xfs/linux-2.6/xfs_lrw.c	2011-08-11 16:51:30.000000000 -0500
+++ linux-2.6.32-SLES11-SP1/fs/xfs/linux-2.6/xfs_lrw.c	2011-08-11 16:52:47.000000000 -0500
@@ -573,6 +573,7 @@
 	size_t			ocount = 0, count;
 	loff_t			pos;
 	int			need_i_mutex;
+	int			unaligned_io = 0;
 
 	XFS_STATS_INC(xs_write_calls);
 
@@ -649,8 +650,12 @@
 			xfs_iunlock(xip, XFS_ILOCK_EXCL|iolock);
 			return XFS_ERROR(-EINVAL);
 		}
+		if ((pos & mp->m_blockmask) ||
+				((pos + count) & mp->m_blockmask))
+			unaligned_io = 1;
 
-		if (!need_i_mutex && (mapping->nrpages || pos > xip->i_size)) {
+		if (!need_i_mutex && (mapping->nrpages ||
+				unaligned_io || pos > xip->i_size)) {
 			xfs_iunlock(xip, XFS_ILOCK_EXCL|iolock);
 			iolock = XFS_IOLOCK_EXCL;
 			need_i_mutex = 1;
@@ -719,7 +724,9 @@
 				goto out_unlock_internal;
 		}
 
-		if (need_i_mutex) {
+		if (unaligned_io)
+			xfs_ioend_wait(xip);
+		else if (need_i_mutex) {
 			/* demote the lock now the cached pages are gone */
 			xfs_ilock_demote(xip, XFS_IOLOCK_EXCL);
 			mutex_unlock(&inode->i_mutex);
