From: Len Brown <len.brown@intel.com>
Subject: intel_idle: add support for Westmere-EX
References: fate#311818,fate#311942
Patch-Mainline: yes
Git-commit: e361b2321279f6c44392bc3e89b4895d68594b67


Signed-off-by: Thomas Renninger <trenn@suse.de>

upstream: ec67a2ba360d4874b1158e6e87fe1e859b0c9117

Signed-off-by: Len Brown <len.brown@intel.com>

diff --git a/drivers/idle/intel_idle.c b/drivers/idle/intel_idle.c
index 6d43bc3..97655ed 100755
--- a/drivers/idle/intel_idle.c
+++ b/drivers/idle/intel_idle.c
@@ -256,6 +256,7 @@ static int intel_idle_probe(void)
 	case 0x1E:	/* Core i7 and i5 Processor - Lynnfield Jasper Forest */
 	case 0x1F:	/* Core i7 and i5 Processor - Nehalem */
 	case 0x2E:	/* Nehalem-EX Xeon */
+	case 0x2F:	/* Westmere-EX Xeon */
 		lapic_timer_reliable_states = (1 << 1);	 /* C1 */
 
 	case 0x25:	/* Westmere */
