From 842e8dfbeb0eced1e5ff7b0423cdf1ed58a7d474 Mon Sep 17 00:00:00 2001
From: Michael Neuling <mikey@neuling.org>
Date: Tue, 8 May 2018 22:42:01 +1000
Subject: [PATCH 2/3] fix hv vs guest mode for p7/p8 case as suggested by suse

References: CVE-2018-3639, bsc#1087082
Patch-mainline: no, under development

Signed-off-by:  Michal Suchanek <msuchanek@suse.de>
---
 arch/powerpc/lib/feature-fixups.c | 15 ++++++++++++---
 1 file changed, 12 insertions(+), 3 deletions(-)

diff --git a/arch/powerpc/lib/feature-fixups.c b/arch/powerpc/lib/feature-fixups.c
index 0dad6557ea49..6a6a360f7383 100644
--- a/arch/powerpc/lib/feature-fixups.c
+++ b/arch/powerpc/lib/feature-fixups.c
@@ -183,12 +183,21 @@ void do_stf_exit_barrier_fixups(enum stf_barrier_type types)
 
 	i = 0;
 	if (types & STF_BARRIER_FALLBACK || types & STF_BARRIER_SYNC_ORI) {
-		instrs[i++] = 0x7db243a6; /* mtsprg 2,r13	*/
-		instrs[i++] = 0x7db142a6; /* mfsprg r13,1	*/
+		if (cpu_has_feature(CPU_FTR_HVMODE)) {
+			instrs[i++] = 0x7db14ba6; /* mtspr 0x131, r13 (HSPRG1) */
+			instrs[i++] = 0x7db04aa6; /* mfspr r13, 0x130 (HSPRG0) */
+		} else {
+			instrs[i++] = 0x7db243a6; /* mtsprg 2,r13	*/
+			instrs[i++] = 0x7db142a6; /* mfsprg r13,1    */
+	        }
 		instrs[i++] = 0x7c0004ac; /* hwsync		*/
 		instrs[i++] = 0xe9ad0000; /* ld r13,0(r13)	*/
 		instrs[i++] = 0x63ff0000; /* ori 31,31,0 speculation barrier */
-		instrs[i++] = 0x7db242a6; /* mfsprg r13,2	*/
+		if (cpu_has_feature(CPU_FTR_HVMODE)) {
+			instrs[i++] = 0x7db14aa6; /* mfspr r13, 0x131 (HSPRG1) */
+		} else {
+			instrs[i++] = 0x7db242a6; /* mfsprg r13,2 */
+		}
 	} else if (types & STF_BARRIER_EIEIO) {
 		instrs[i++] = 0x7e0006ac; /* eieio + bit 6 hint */
 	}
-- 
2.12.3

