From: Yinghai Lu <yinghai@kernel.org>
Subject: intel-iommu: Don't complain that ACPI_DMAR_SCOPE_TYPE_IOAPIC is not supported
References: bnc#579639, FATE#306952
Patch-mainline: not yet

Signed-off-by: Yinghai Lu <yinghai@kernel.org>
Signed-off-by: David Woodhouse <David.Woodhouse@intel.com>
Signed-off-by: Rafael J. Wysocki <rjw@suse.de>
---
 drivers/pci/dmar.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

Index: linux-2.6.32-SLE11-SP1/drivers/pci/dmar.c
===================================================================
--- linux-2.6.32-SLE11-SP1.orig/drivers/pci/dmar.c
+++ linux-2.6.32-SLE11-SP1/drivers/pci/dmar.c
@@ -130,9 +130,10 @@ static int __init dmar_parse_dev_scope(v
 		if (scope->entry_type == ACPI_DMAR_SCOPE_TYPE_ENDPOINT ||
 		    scope->entry_type == ACPI_DMAR_SCOPE_TYPE_BRIDGE)
 			(*cnt)++;
-		else
+		else if (scope->entry_type != ACPI_DMAR_SCOPE_TYPE_IOAPIC) {
 			printk(KERN_WARNING PREFIX
-				"Unsupported device scope\n");
+			       "Unsupported device scope\n");
+		}
 		start += scope->length;
 	}
 	if (*cnt == 0)
