From: Martin Schwidefsky <schwidefsky@de.ibm.com>
Patch-mainline: Not yet, under development
Subject: s390/spinlock: add gmb memory barrier

Signed-off-by: Martin Schwidefsky <schwidefsky@de.ibm.com>
Signed-off-by: Jiri Kosina <jkosina@suse.cz>
---
 arch/s390/include/asm/system.h |    9 +++++++++
 1 file changed, 9 insertions(+)

--- a/arch/s390/include/asm/system.h
+++ b/arch/s390/include/asm/system.h
@@ -126,6 +126,8 @@
 
 #define nop() asm volatile("nop")
 
+#include <asm/alternative.h>
+
 /*
  * Force strict CPU ordering.
  * And yes, this is required on UP too when we're talking
@@ -149,6 +151,12 @@
 #define smp_mb__before_clear_bit()     smp_mb()
 #define smp_mb__after_clear_bit()      smp_mb()
 
+static inline void gmb(void)
+{
+	asm volatile(
+		ALTERNATIVE("", ".long 0xb2e8f000", 49)
+		: : : "memory");
+}
 
 #define set_mb(var, value)      do { var = value; mb(); } while (0)
 
