From: Thomas Renninger <trenn@suse.de>
Subject: x86 backport fixup: Add missing include
References: fate#309722, fate#309723, fate#309780, fate#309781
Patch-Mainline: yes

Signed-off-by: Thomas Renninger <trenn@suse.de>

---
 arch/x86/pci/amd_bus.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.32-SLE11-SP2/arch/x86/pci/amd_bus.c
===================================================================
--- linux-2.6.32-SLE11-SP2.orig/arch/x86/pci/amd_bus.c
+++ linux-2.6.32-SLE11-SP2/arch/x86/pci/amd_bus.c
@@ -4,9 +4,9 @@
 #include <linux/cpu.h>
 #include <asm/amd_nb.h>
 #include <asm/pci_x86.h>
+#include <asm/pci-direct.h>
 
 #ifdef CONFIG_X86_64
-#include <asm/pci-direct.h>
 #include <asm/mpspec.h>
 #include <linux/cpumask.h>
 #endif
