From fc4833b5b752a2d1aa0502386aab2d6a120db8e5 Mon Sep 17 00:00:00 2001
From: Lidong Zhong <lzhong@suse.com>
Date: Thu, 24 May 2018 11:13:48 +0800
Subject: [PATCH] x86/kaiser: symbol kaiser_set_shadow_pgd() exported with non
 GPL
References: bsc#1092813
Patch-mainline: No, different upstream implementation

Signed-off-by: Lidong Zhong <lzhong@suse.com>
---
 arch/x86/mm/kaiser.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/mm/kaiser.c b/arch/x86/mm/kaiser.c
index 513b24b67efd..a25fac326686 100644
--- a/arch/x86/mm/kaiser.c
+++ b/arch/x86/mm/kaiser.c
@@ -451,7 +451,7 @@ pgd_t kaiser_set_shadow_pgd(pgd_t *pgdp, pgd_t pgd)
 	}
 	return pgd;
 }
-EXPORT_SYMBOL_GPL(kaiser_set_shadow_pgd);
+EXPORT_SYMBOL(kaiser_set_shadow_pgd);
 
 void kaiser_setup_pcid(void)
 {
-- 
2.13.6

