From 3661facd402932aa5ff2fc375cc7389f3175ba82 Mon Sep 17 00:00:00 2001
From: "Lee, Chun-Yi" <jlee@suse.com>
Date: Tue, 18 Aug 2015 12:11:07 +0800
Subject: [PATCH] PM / hibernate: Force verify hibernation signature in signed
 modules environment

Patch-mainline: Never, unless BSD-style securelevel accepted by upstream
References: fate#316350

Signed-off-by: Lee, Chun-Yi <jlee@suse.com>
---
 kernel/power/hibernate.c |   25 +++++++++++++++++++++++++
 1 file changed, 25 insertions(+)

--- a/kernel/power/hibernate.c
+++ b/kernel/power/hibernate.c
@@ -28,6 +28,7 @@
 #include <linux/syscore_ops.h>
 #include <linux/ctype.h>
 #include <linux/genhd.h>
+#include <linux/module.h>
 
 #include "power.h"
 
@@ -645,6 +646,14 @@ int hibernate(void)
 
 	set_hibernation_key_regen_flag = false;
 
+	if (secure_modules()) {
+#ifdef CONFIG_HIBERNATE_VERIFICATION
+		sigenforce = 1;
+#else
+		return -EPERM;
+#endif
+	}
+
 	lock_system_sleep();
 	/* The snapshot device should not be opened while we're running */
 	if (!atomic_add_unless(&snapshot_device_available, -1, 0)) {
@@ -741,6 +750,14 @@ static int software_resume(void)
 	if (noresume)
 		return 0;
 
+	if (secure_modules()) {
+#ifdef CONFIG_HIBERNATE_VERIFICATION
+               sigenforce = 1;
+#else
+               return 0;
+#endif
+	}
+
 	/*
 	 * name_to_dev_t() below takes a sysfs buffer mutex when sysfs
 	 * is configured into the kernel. Since the regular hibernate
@@ -938,6 +955,14 @@ static ssize_t disk_store(struct kobject
 	char *p;
 	int mode = HIBERNATION_INVALID;
 
+	if (secure_modules()) {
+#ifdef CONFIG_HIBERNATE_VERIFICATION
+		sigenforce = 1;
+#else
+		return -EPERM;
+#endif
+	}
+
 	p = memchr(buf, '\n', n);
 	len = p ? p - buf : n;
 
