From 03ec122d0d3fcea6a284cfa48480c4aa2c74c45b Mon Sep 17 00:00:00 2001
From: Lidong Zhong <lzhong@suse.com>
Date: Thu, 24 May 2018 10:59:51 +0800
Subject: [PATCH] x86/kaiser: export symbol kaiser_set_shadow_pgd()
References: bsc#1092813
Patch-mainline: No, different upstream implementation

Signed-off-by: Lidong Zhong <lzhong@suse.com>
---
 arch/x86/mm/kaiser.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/x86/mm/kaiser.c b/arch/x86/mm/kaiser.c
index a0d0bb47b6bc..36e0d67bb046 100644
--- a/arch/x86/mm/kaiser.c
+++ b/arch/x86/mm/kaiser.c
@@ -451,6 +451,7 @@ pgd_t kaiser_set_shadow_pgd(pgd_t *pgdp, pgd_t pgd)
 	}
 	return pgd;
 }
+EXPORT_SYMBOL(kaiser_set_shadow_pgd);
 
 void kaiser_setup_pcid(void)
 {
-- 
2.13.6

