From 7f1f5abfc05fed1c2ad99f5bb040f247fe8e171b Mon Sep 17 00:00:00 2001
From: Lidong Zhong <lzhong@suse.com>
Date: Fri, 27 Apr 2018 10:16:19 +0800
Patch-mainline: Never, different upstream implementation
References: bsc#1090630
Subject: [PATCH] x86/kaiser: export symbol kaiser_set_shadow_pgd(bsc#1090630)

Signed-off-by: Lidong Zhong <lzhong@suse.com>
---
 arch/x86/mm/kaiser.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/x86/mm/kaiser.c b/arch/x86/mm/kaiser.c
index 00f2d41ac8e8..513b24b67efd 100644
--- a/arch/x86/mm/kaiser.c
+++ b/arch/x86/mm/kaiser.c
@@ -451,6 +451,7 @@ pgd_t kaiser_set_shadow_pgd(pgd_t *pgdp, pgd_t pgd)
 	}
 	return pgd;
 }
+EXPORT_SYMBOL_GPL(kaiser_set_shadow_pgd);
 
 void kaiser_setup_pcid(void)
 {
-- 
2.13.6

