From: Jeff Garzik <jeff@garzik.org>
Date: Fri, 14 May 2010 17:23:37 -0400
Subject: [PATCH] ahci_platform: properly set up EM messaging
Git-commit: 55787183ade44c4f826f581a068f52a1a80c6a2e
References: FATE#311692
Patch-Mainline: 2.6.34

Signed-off-by: Jeff Garzik <jgarzik@redhat.com>
Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 drivers/ata/ahci_platform.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/ata/ahci_platform.c b/drivers/ata/ahci_platform.c
index 19bfd1d..5e11b16 100644
--- a/drivers/ata/ahci_platform.c
+++ b/drivers/ata/ahci_platform.c
@@ -125,7 +125,7 @@ static int __init ahci_probe(struct platform_device *pdev)
 
 		/* set enclosure management message type */
 		if (ap->flags & ATA_FLAG_EM)
-			ap->em_message_type = ahci_em_messages;
+			ap->em_message_type = hpriv->em_msg_type;
 
 		/* disabled/not-implemented port */
 		if (!(hpriv->port_map & (1 << i)))
-- 
1.6.0.2

