From: Sergei Trofimovich <slyfox@gentoo.org>
Date: Fri, 20 May 2011 20:20:31 +0000
Patch-mainline: yes
References: FATE#306586
Subject: [PATCH] btrfs: typo: 'btrfS' -> 'btrfs'

Signed-off-by: Sergei Trofimovich <slyfox@gentoo.org>
Signed-off-by: Chris Mason <chris.mason@oracle.com>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/dir-item.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.32-SLE11-SP2/fs/btrfs/dir-item.c
===================================================================
--- linux-2.6.32-SLE11-SP2.orig/fs/btrfs/dir-item.c
+++ linux-2.6.32-SLE11-SP2/fs/btrfs/dir-item.c
@@ -438,7 +438,7 @@ int verify_dir_item(struct btrfs_root *r
 		namelen = XATTR_NAME_MAX;
 
 	if (btrfs_dir_name_len(leaf, dir_item) > namelen) {
-		printk(KERN_CRIT "btrfS: invalid dir item name len: %u\n",
+		printk(KERN_CRIT "btrfs: invalid dir item name len: %u\n",
 		       (unsigned)btrfs_dir_data_len(leaf, dir_item));
 		return 1;
 	}
