From: Chris Mason <chris.mason@oracle.com>
Date: Mon, 20 Feb 2012 22:14:55 -0500
References: FATE#306586
Git-commit: 16780cabb877dbd0c8c5e9ff9bdebd6c5bdd1a7b
Subject: [PATCH] Btrfs: add extra sanity checks on the path names in
 btrfs_mksubvol
Patch-mainline: v3.3-rc7

Signed-off-by: Chris Mason <chris.mason@oracle.com>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/ioctl.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/fs/btrfs/ioctl.c b/fs/btrfs/ioctl.c
index e9bdb8b..05446f7 100644
--- a/fs/btrfs/ioctl.c
+++ b/fs/btrfs/ioctl.c
@@ -1333,6 +1333,12 @@ static noinline int btrfs_ioctl_snap_create_transid(struct file *file,
 		goto out;
 	}
 
+	if (name[0] == '.' &&
+	   (namelen == 1 || (name[1] == '.' && namelen == 2))) {
+		ret = -EEXIST;
+		goto out;
+	}
+
 	if (subvol) {
 		ret = btrfs_mksubvol(&file->f_path, name, namelen,
 				     NULL, transid, readonly);
-- 
1.7.6.233.gd79bc

