From: Chris Mason <chris.mason@fusionio.com>
Date: Wed, 25 Jul 2012 19:21:10 -0400
Patch-mainline: 3.6
Git-commit: b24baf6917a376420d535548e1f88744028bcf24
References: FATE#312888
Subject: [PATCH] Btrfs: uninit variable fixes in send/receive

Signed-off-by: Chris Mason <chris.mason@fusionio.com>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/send.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/fs/btrfs/send.c
+++ b/fs/btrfs/send.c
@@ -2676,7 +2676,7 @@ static int process_recorded_refs(struct
 	struct ulist_iterator uit;
 	struct ulist_node *un;
 	struct fs_path *valid_path = NULL;
-	u64 ow_inode;
+	u64 ow_inode = 0;
 	u64 ow_gen;
 	int did_overwrite = 0;
 	int is_orphan = 0;
@@ -3553,7 +3553,7 @@ static int send_write(struct send_ctx *s
 	int ret = 0;
 	struct fs_path *p;
 	loff_t pos = offset;
-	int readed;
+	int readed = 0;
 	mm_segment_t old_fs;
 
 	p = fs_path_alloc(sctx);
