From: David Sterba <dsterba@suse.cz>
Date: Mon, 11 Mar 2013 17:07:45 +0000
Patch-mainline: 3.10
Git-commit: bbece8a3f00a02bbfc63531651fd70ef56c5e916
References: FATE#312888
Subject: [PATCH] btrfs: merge save_error_info helpers into one

Signed-off-by: David Sterba <dsterba@suse.cz>
Signed-off-by: Josef Bacik <jbacik@fusionio.com>
---
 fs/btrfs/super.c |   11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

--- a/fs/btrfs/super.c
+++ b/fs/btrfs/super.c
@@ -92,7 +92,7 @@ static const char *btrfs_decode_error(in
 	return errstr;
 }
 
-static void __save_error_info(struct btrfs_fs_info *fs_info)
+static void save_error_info(struct btrfs_fs_info *fs_info)
 {
 	/*
 	 * today we only save the error info into ram.  Long term we'll
@@ -101,15 +101,6 @@ static void __save_error_info(struct btr
 	set_bit(BTRFS_FS_STATE_ERROR, &fs_info->fs_state);
 }
 
-/* NOTE:
- *	We move write_super stuff at umount in order to avoid deadlock
- *	for umount hold all lock.
- */
-static void save_error_info(struct btrfs_fs_info *fs_info)
-{
-	__save_error_info(fs_info);
-}
-
 /* btrfs handle error by forcing the filesystem readonly */
 static void btrfs_handle_error(struct btrfs_fs_info *fs_info)
 {
