From: Al Viro <viro@zeniv.linux.org.uk>
Date: Wed, 16 Nov 2011 16:06:09 -0500
Patch-mainline: 3.2
Git-commit: 8d514bbf37eecf0a3e309284728637816a36764b
Subject: [PATCH] btrfs: fix double mntput() in mount_subvol()

Signed-off-by: Al Viro <viro@zeniv.linux.org.uk>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/super.c |    1 -
 1 file changed, 1 deletion(-)

--- a/fs/btrfs/super.c
+++ b/fs/btrfs/super.c
@@ -861,7 +861,6 @@ static struct dentry *mount_subvol(const
 
 	if (!is_subvolume_inode(nd.path.dentry->d_inode)) {
 		path_put(&nd.path);
-		mntput(mnt);
 		error = -EINVAL;
 		printk(KERN_ERR "btrfs: '%s' is not a valid subvolume\n",
 				subvol_name);
