From: David Sterba <dsterba@suse.cz>
Date: Fri, 26 Apr 2013 12:56:04 +0000
Patch-mainline: 3.10
Git-commit: 34c2b29079ba505a1803fa620b5f116f07e48258
References: FATE#312888
Subject: [PATCH] btrfs: handle errors returned from get_tree_block_key

Signed-off-by: David Sterba <dsterba@suse.cz>
Reviewed-by: Zach Brown <zab@redhat.com>
Signed-off-by: Josef Bacik <jbacik@fusionio.com>
---
 fs/btrfs/relocation.c |   12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

--- a/fs/btrfs/relocation.c
+++ b/fs/btrfs/relocation.c
@@ -2862,7 +2862,7 @@ int relocate_tree_blocks(struct btrfs_tr
 	path = btrfs_alloc_path();
 	if (!path) {
 		err = -ENOMEM;
-		goto out_path;
+		goto out_free_blocks;
 	}
 
 	rb_node = rb_first(blocks);
@@ -2876,8 +2876,11 @@ int relocate_tree_blocks(struct btrfs_tr
 	rb_node = rb_first(blocks);
 	while (rb_node) {
 		block = rb_entry(rb_node, struct tree_block, rb_node);
-		if (!block->key_ready)
-			get_tree_block_key(rc, block);
+		if (!block->key_ready) {
+			err = get_tree_block_key(rc, block);
+			if (err)
+				goto out_free_path;
+		}
 		rb_node = rb_next(rb_node);
 	}
 
@@ -2904,8 +2907,9 @@ int relocate_tree_blocks(struct btrfs_tr
 out:
 	err = finish_pending_nodes(trans, rc, path, err);
 
+out_free_path:
 	btrfs_free_path(path);
-out_path:
+out_free_blocks:
 	free_block_list(blocks);
 	return err;
 }
