From: Al Viro <viro@zeniv.linux.org.uk>
Date: Wed, 16 Nov 2011 16:06:09 -0500
Patch-mainline: 3.2
Git-commit: 8d514bbf37eecf0a3e309284728637816a36764b
Subject: [PATCH] btrfs: fix double mntput() in mount_subvol()

Signed-off-by: Al Viro <viro@zeniv.linux.org.uk>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/super.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/fs/btrfs/super.c b/fs/btrfs/super.c
index 8bd9d6d..969a774 100644
--- a/fs/btrfs/super.c
+++ b/fs/btrfs/super.c
@@ -861,7 +861,6 @@ static struct dentry *mount_subvol(const char *subvol_name, int flags,
 
 	if (!is_subvolume_inode(path.dentry->d_inode)) {
 		path_put(&path);
-		mntput(mnt);
 		error = -EINVAL;
 		printk(KERN_ERR "btrfs: '%s' is not a valid subvolume\n",
 				subvol_name);
-- 
1.8.0

