From: David Sterba <dsterba@suse.cz>
Date: Mon, 12 Nov 2012 17:19:24 +0100
Patch-mainline: pending
References: bnc#752067
Subject: [PATCH] btrfs: continue after abort during snapshot drop

Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/transaction.c |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/fs/btrfs/transaction.c b/fs/btrfs/transaction.c
index 04bbfb1..e3027dd 100644
--- a/fs/btrfs/transaction.c
+++ b/fs/btrfs/transaction.c
@@ -1750,7 +1750,10 @@ int btrfs_clean_old_snapshots(struct btrfs_root *root)
 			ret = btrfs_drop_snapshot(root, NULL, 0, 0);
 		else
 			ret =btrfs_drop_snapshot(root, NULL, 1, 0);
-		BUG_ON(ret < 0);
+		/*
+		 * Continue after a transaction abort during drop
+		 */
+		BUG_ON(ret < 0 && ret != -EROFS);
 	}
 	return 0;
 }
-- 
1.7.9

