From: David Sterba <dsterba@suse.com>
Date: Mon, 11 Apr 2016 18:40:08 +0200
Patch-mainline: 4.7
Git-commit: 153519559a39725c5a45269256fec0efb81bcd1f
References: bsc#1012452
Subject: [PATCH] btrfs: clone: use vmalloc only as fallback for
 nodesize bufer

Signed-off-by: David Sterba <dsterba@suse.com>
---
 fs/btrfs/ioctl.c |   13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

--- a/fs/btrfs/ioctl.c
+++ b/fs/btrfs/ioctl.c
@@ -3562,13 +3562,16 @@ static int btrfs_clone(struct inode *src
 	u64 last_dest_end = destoff;
 
 	ret = -ENOMEM;
-	buf = vmalloc(root->nodesize);
-	if (!buf)
-		return ret;
+	buf = kmalloc(root->nodesize, GFP_KERNEL | __GFP_NOWARN);
+	if (!buf) {
+		buf = vmalloc(root->nodesize);
+		if (!buf)
+			return ret;
+	}
 
 	path = btrfs_alloc_path();
 	if (!path) {
-		vfree(buf);
+		kvfree(buf);
 		return ret;
 	}
 
@@ -3874,7 +3877,7 @@ process_slot:
 
 out:
 	btrfs_free_path(path);
-	vfree(buf);
+	kvfree(buf);
 	return ret;
 }
 
