From: Miao Xie <miaox@cn.fujitsu.com>
Date: Wed, 15 May 2013 07:48:17 +0000
Patch-mainline: 3.10
Git-commit: 061594ef171a5ba52b5786688ae766907b0bda2b
References: FATE#312888
Subject: [PATCH] Btrfs: pause the space balance when remounting to R/O

Signed-off-by: Miao Xie <miaox@cn.fujitsu.com>
Signed-off-by: Josef Bacik <jbacik@fusionio.com>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/super.c |    1 +
 1 file changed, 1 insertion(+)

--- a/fs/btrfs/super.c
+++ b/fs/btrfs/super.c
@@ -1265,6 +1265,7 @@ static int btrfs_remount(struct super_bl
 
 		btrfs_dev_replace_suspend_for_unmount(fs_info);
 		btrfs_scrub_cancel(fs_info);
+		btrfs_pause_balance(fs_info);
 
 		ret = btrfs_commit_super(root);
 		if (ret)
