From: Wang Shilong <wangsl-fnst@cn.fujitsu.com>
Date: Tue, 16 Apr 2013 10:22:23 +0000
Patch-mainline: 3.10
Git-commit: ccf7f29d1a23696ca69a981adbf7eda2e13c8635
Subject: [PATCH] Btrfs: remove unused variable in the
 iterate_extent_inodes()

Signed-off-by: Wang Shilong <wangsl-fnst@cn.fujitsu.com>
Signed-off-by: Josef Bacik <jbacik@fusionio.com>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/backref.c |    2 --
 1 file changed, 2 deletions(-)

--- a/fs/btrfs/backref.c
+++ b/fs/btrfs/backref.c
@@ -1464,8 +1464,6 @@ int iterate_extent_inodes(struct btrfs_f
 				iterate_extent_inodes_t *iterate, void *ctx)
 {
 	int ret;
-	struct list_head data_refs = LIST_HEAD_INIT(data_refs);
-	struct list_head shared_refs = LIST_HEAD_INIT(shared_refs);
 	struct btrfs_trans_handle *trans;
 	struct ulist *refs = NULL;
 	struct ulist *roots = NULL;
