From: David Sterba <dsterba@suse.cz>
Date: Wed, 20 Mar 2013 14:29:47 +0000
Patch-mainline: pending
Subject: [PATCH] btrfs: cover more error codes in btrfs_decode_error

Signed-off-by: David Sterba <dsterba@suse.cz>
Signed-off-by: Josef Bacik <jbacik@fusionio.com>
---
 fs/btrfs/super.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/fs/btrfs/super.c b/fs/btrfs/super.c
index f4da2d5..984816f 100644
--- a/fs/btrfs/super.c
+++ b/fs/btrfs/super.c
@@ -81,6 +81,12 @@ static const char *btrfs_decode_error(int errno)
 	case -EEXIST:
 		errstr = "Object already exists";
 		break;
+	case -ENOSPC:
+		errstr = "No space left";
+		break;
+	case -ENOENT:
+		errstr = "No such entry";
+		break;
 	}
 
 	return errstr;
-- 
1.8.2

