From: Jens Axboe <jens.axboe@oracle.com>
Date: Thu, 3 Dec 2009 21:06:43 +0100
Subject: [PATCH] cfq-iosched: fix compile problem with !CONFIG_CGROUP
Git-commit: 2f5ea47712489a9d2d3cb832eb06062e4e64e0ec
References: FATE#311054
Patch-Mainline: 2.6.33

Signed-off-by: Jens Axboe <jens.axboe@oracle.com>
Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 block/blk-cgroup.h |   10 ++++++++++
 1 files changed, 10 insertions(+), 0 deletions(-)

diff --git a/block/blk-cgroup.h b/block/blk-cgroup.h
index b24ab71..257dc6a 100644
--- a/block/blk-cgroup.h
+++ b/block/blk-cgroup.h
@@ -15,6 +15,8 @@
 
 #include <linux/cgroup.h>
 
+#ifdef CONFIG_BLK_CGROUP
+
 struct blkio_cgroup {
 	struct cgroup_subsys_state css;
 	unsigned int weight;
@@ -41,6 +43,13 @@ struct blkio_group {
 	unsigned long sectors;
 };
 
+#else
+
+struct blkio_group {
+};
+
+#endif
+
 #define BLKIO_WEIGHT_MIN	100
 #define BLKIO_WEIGHT_MAX	1000
 #define BLKIO_WEIGHT_DEFAULT	500
@@ -69,6 +78,7 @@ extern struct blkio_group *blkiocg_lookup_group(struct blkio_cgroup *blkcg,
 void blkiocg_update_blkio_group_stats(struct blkio_group *blkg,
 			unsigned long time, unsigned long sectors);
 #else
+struct cgroup;
 static inline struct blkio_cgroup *
 cgroup_to_blkio_cgroup(struct cgroup *cgroup) { return NULL; }
 
-- 
1.6.0.2

