From: Jeff Layton <jlayton@redhat.com>
Date: Fri, 8 Oct 2010 12:20:12 -0400
Subject: cifs: initialize tlink_tree_lock and tlink_tree
References: FATE#311695
Git-commit: 0dd12c21951e085357046b68c3a108273062d2aa
Patch-mainline: v2.6.37-rc1

Signed-off-by: Jeff Layton <jlayton@redhat.com>
Signed-off-by: Steve French <sfrench@us.ibm.com>
Signed-off-by: Suresh Jayaraman <sjayaraman@suse.de>
---

Index: linux-2.6.32-SLE11-SP2/fs/cifs/cifsfs.c
===================================================================
 fs/cifs/cifsfs.c |    3 +++
 1 file changed, 3 insertions(+)

--- linux-2.6.32-SLE11-SP2.orig/fs/cifs/cifsfs.c
+++ linux-2.6.32-SLE11-SP2/fs/cifs/cifsfs.c
@@ -99,6 +99,9 @@ cifs_read_super(struct super_block *sb,
 	if (cifs_sb == NULL)
 		return -ENOMEM;
 
+	spin_lock_init(&cifs_sb->tlink_tree_lock);
+	INIT_RADIX_TREE(&cifs_sb->tlink_tree, GFP_KERNEL);
+
 #ifdef CONFIG_CIFS_DFS_UPCALL
 	/* copy mount params to sb for use in submounts */
 	/* BB: should we move this after the mount so we
