From: Suresh Jayaraman <sjayaraman@suse.de>
Date: Thu, 5 Aug 2010 18:52:21 +0530
Subject: cifs: update README to include details about 'fsc' option
References: FATE#311695
Git-commit: 3694b91a59c3bc784735574878d36652dfd623c2
Patch-mainline: v2.6.36-rc1

Signed-off-by: Suresh Jayaraman <sjayaraman@suse.de>
Signed-off-by: Steve French <sfrench@us.ibm.com>
---
 fs/cifs/README |   10 ++++++++++
 1 files changed, 10 insertions(+), 0 deletions(-)

Index: linux-2.6.32-SLE11-SP2/fs/cifs/README
===================================================================
--- linux-2.6.32-SLE11-SP2.orig/fs/cifs/README
+++ linux-2.6.32-SLE11-SP2/fs/cifs/README
@@ -301,6 +301,16 @@ A partial list of the supported mount op
   gid		Set the default gid for inodes (similar to above).
   file_mode     If CIFS Unix extensions are not supported by the server
 		this overrides the default mode for file inodes.
+  fsc		Enable local disk caching using FS-Cache (off by default). This
+  		option could be useful to improve performance on a slow link,
+		heavily loaded server and/or network where reading from the
+		disk is faster than reading from the server (over the network).
+		This could also impact scalability positively as the
+		number of calls to the server are reduced. However, local
+		caching is not suitable for all workloads for e.g. read-once
+		type workloads. So, you need to consider carefully your
+		workload/scenario before using this option. Currently, local
+		disk caching is functional for CIFS files opened as read-only.
   dir_mode      If CIFS Unix extensions are not supported by the server 
 		this overrides the default mode for directory inodes.
   port		attempt to contact the server on this tcp port, before
