From: Steve French <sfrench@us.ibm.com>
Date: Wed, 19 Jan 2011 17:53:44 +0000
Subject: [CIFS] Update cifs version number
References: FATE#311695
Git-commit: 1cd3508d5eab6a88fa643119cedd34b04215cffe
Patch-mainline: v2.6.38-rc2

Signed-off-by: Steve French <sfrench@us.ibm.com>
Signed-off-by: Suresh Jayaraman <sjayaraman@suse.de>
---
 fs/cifs/cifsfs.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

Index: linux-2.6.32-SLE11-SP2/fs/cifs/cifsfs.h
===================================================================
--- linux-2.6.32-SLE11-SP2.orig/fs/cifs/cifsfs.h
+++ linux-2.6.32-SLE11-SP2/fs/cifs/cifsfs.h
@@ -115,5 +115,5 @@ extern void cifs_destroy_tlinks_queue(vo
 extern const struct export_operations cifs_export_ops;
 #endif /* EXPERIMENTAL */
 
-#define CIFS_VERSION   "1.68"
+#define CIFS_VERSION   "1.69"
 #endif				/* _CIFSFS_H */
