From: Steve French <sfrench@us.ibm.com>
Date: Mon, 31 Jan 2011 21:56:35 +0000
Subject: [CIFS] Update cifs minor version
References: FATE#311695
Git-commit: cab6958da0094e36a098751f844409fc9ee26251
Patch-mainline: v2.6.38-rc4

Signed-off-by: Steve French <sfrench@us.ibm.com>
Signed-off-by: Suresh Jayaraman <sjayaraman@suse.de>
---
 fs/cifs/cifsfs.h |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

Index: linux-2.6.32-SLE11-SP2/fs/cifs/cifsfs.h
===================================================================
--- linux-2.6.32-SLE11-SP2.orig/fs/cifs/cifsfs.h
+++ linux-2.6.32-SLE11-SP2/fs/cifs/cifsfs.h
@@ -118,5 +118,5 @@ extern void cifs_destroy_echo_queue(void
 extern const struct export_operations cifs_export_ops;
 #endif /* EXPERIMENTAL */
 
-#define CIFS_VERSION   "1.69"
+#define CIFS_VERSION   "1.70"
 #endif				/* _CIFSFS_H */
