From: Jiri Bohac <jbohac@suse.cz>
Subject: ebt_vlan: add a dummy debug option
References: bnc#720451
Patch-mainline: Never

diff --git a/net/bridge/netfilter/ebt_vlan.c b/net/bridge/netfilter/ebt_vlan.c
index eae67bf..546a079 100644
--- a/net/bridge/netfilter/ebt_vlan.c
+++ b/net/bridge/netfilter/ebt_vlan.c
@@ -28,6 +28,8 @@
 
 #define MODULE_VERS "0.6"
 
+static int debug;
+module_param(debug, int, 0); /* dummy parameter for backwards compatibility */
 MODULE_AUTHOR("Nick Fedchik <nick@fedchik.org.ua>");
 MODULE_DESCRIPTION("Ebtables: 802.1Q VLAN tag match");
 MODULE_LICENSE("GPL");
