From: David Sterba <dsterba@suse.cz>
Date: Wed, 23 Nov 2011 01:11:06 +0100
Patch-mainline: pending
References: FATE#306586
Subject: [PATCH] fs: FIEMAP: add flag for compressed extent

Signed-off-by: David Sterba <dsterba@suse.cz>
---
 include/linux/fiemap.h |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/include/linux/fiemap.h b/include/linux/fiemap.h
index d830747..57631db 100644
--- a/include/linux/fiemap.h
+++ b/include/linux/fiemap.h
@@ -49,6 +49,8 @@ struct fiemap {
 						    * Sets EXTENT_UNKNOWN. */
 #define FIEMAP_EXTENT_ENCODED		0x00000008 /* Data can not be read
 						    * while fs is unmounted */
+#define FIEMAP_EXTENT_DATA_COMPRESSED   0x00000040 /* Data is compressed by fs.
+						    * Sets EXTENT_NO_BYPASS. */
 #define FIEMAP_EXTENT_DATA_ENCRYPTED	0x00000080 /* Data is encrypted by fs.
 						    * Sets EXTENT_NO_BYPASS. */
 #define FIEMAP_EXTENT_NOT_ALIGNED	0x00000100 /* Extent offsets may not be
-- 
1.7.7.3

