From: Miroslav Benes <mbenes@suse.cz>
Subject: kgr: add kgraft annotation to hwrng kthread
Patch-mainline: no, SLE12 specific
References: fate#313296

Signed-off-by: Miroslav Benes <mbenes@suse.cz>
Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 drivers/char/random.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

--- a/drivers/char/random.c
+++ b/drivers/char/random.c
@@ -1613,8 +1613,9 @@ void add_hwgenerator_randomness(const ch
 	 * We'll be woken up again once below random_write_wakeup_thresh,
 	 * or when the calling thread is about to terminate.
 	 */
-	wait_event_interruptible(random_write_wait, kthread_should_stop() ||
-				 ENTROPY_BITS(&input_pool) <= random_write_wakeup_thresh);
+	wait_event_interruptible(random_write_wait, ({ kgr_task_safe(current);
+				 kthread_should_stop() ||
+				 ENTROPY_BITS(&input_pool) <= random_write_wakeup_thresh; }));
 	mix_pool_bytes(poolp, buffer, count, NULL);
 	credit_entropy_bits(poolp, entropy);
 }
