From: Jiri Slaby <jslaby@suse.cz>
Date: Thu, 30 Jan 2014 11:04:10 +0100
Subject: kgr: sched.h, introduce kgr_task_safe helper
Patch-mainline: submitted for review
References: fate#313296

To be used from some kthreads.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
Cc: Steven Rostedt <rostedt@goodmis.org>
Cc: Frederic Weisbecker <fweisbec@gmail.com>
Cc: Ingo Molnar <mingo@redhat.com>
---
 include/linux/sched.h |    9 +++++++++
 1 file changed, 9 insertions(+)

--- a/include/linux/sched.h
+++ b/include/linux/sched.h
@@ -2814,6 +2814,15 @@ static inline void mm_init_owner(struct
 }
 #endif /* CONFIG_MM_OWNER */
 
+#if IS_ENABLED(CONFIG_KGRAFT)
+static inline void kgr_task_safe(struct task_struct *p)
+{
+	clear_bit(0, &task_thread_info(p)->kgr_in_progress);
+}
+#else
+static inline void kgr_task_safe(struct task_struct *p) { }
+#endif /* IS_ENABLED(CONFIG_KGRAFT) */
+
 static inline unsigned long task_rlimit(const struct task_struct *tsk,
 		unsigned int limit)
 {
