From: Avi Kivity <avi@redhat.com>
Subject: [PATCH] KVM: Drop kvm_{load,put}_guest_fpu() exports
References: FATE#311768
Git-commit e5bb40251a920cdd9d12c569c6aab0bdd0279e4e
Patch-mainline: v2.6.34

Not used anymore.

Signed-off-by: Avi Kivity <avi@redhat.com>
Signed-off-by: Marcelo Tosatti <mtosatti@redhat.com>
Acked-by: Bruce Rogers <brogers@suse.com>
---
 arch/x86/kvm/x86.c |    2 --
 1 file changed, 2 deletions(-)

Index: b/arch/x86/kvm/x86.c
===================================================================
--- a/arch/x86/kvm/x86.c
+++ b/arch/x86/kvm/x86.c
@@ -5071,7 +5071,6 @@ void kvm_load_guest_fpu(struct kvm_vcpu
 	kvm_fx_save(&vcpu->arch.host_fx_image);
 	kvm_fx_restore(&vcpu->arch.guest_fx_image);
 }
-EXPORT_SYMBOL_GPL(kvm_load_guest_fpu);
 
 void kvm_put_guest_fpu(struct kvm_vcpu *vcpu)
 {
@@ -5084,7 +5083,6 @@ void kvm_put_guest_fpu(struct kvm_vcpu *
 	++vcpu->stat.fpu_reload;
 	set_bit(KVM_REQ_DEACTIVATE_FPU, &vcpu->requests);
 }
-EXPORT_SYMBOL_GPL(kvm_put_guest_fpu);
 
 void kvm_arch_vcpu_free(struct kvm_vcpu *vcpu)
 {
