From: Avi Kivity <avi@redhat.com>
Subject: [PATCH] KVM: Set cr0.et when the guest writes cr0
References: FATE#311768
Git-commit: f9a48e6a18c210c4bf34769fa05ede250575c6a1
Patch-mainline: v2.6.34

Follow the hardware.

Signed-off-by: Avi Kivity <avi@redhat.com>
Acked-by: Bruce Rogers <brogers@suse.com>
---
 arch/x86/kvm/x86.c |    2 ++
 1 file changed, 2 insertions(+)

Index: b/arch/x86/kvm/x86.c
===================================================================
--- a/arch/x86/kvm/x86.c
+++ b/arch/x86/kvm/x86.c
@@ -299,6 +299,8 @@ out:
 
 void kvm_set_cr0(struct kvm_vcpu *vcpu, unsigned long cr0)
 {
+	cr0 |= X86_CR0_ET;
+
 	if (cr0 & CR0_RESERVED_BITS) {
 		kvm_inject_gp(vcpu, 0);
 		return;
