From: Mel Gorman <mgorman@suse.de>
Date: Tue, 17 May 2011 15:06:03 +0100
Subject: [PATCH] mm: compaction: Update caller of migrate_pages in ia64 for new API
References: Compaction/reclaim fixes (fate#311931)
Patch-mainline: No

Signed-off-by: Mel Gorman <mgorman@suse.de>
---
 arch/ia64/kernel/cpe_migrate.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/arch/ia64/kernel/cpe_migrate.c b/arch/ia64/kernel/cpe_migrate.c
index 4ccb277..dd21ff0 100644
--- a/arch/ia64/kernel/cpe_migrate.c
+++ b/arch/ia64/kernel/cpe_migrate.c
@@ -178,7 +178,7 @@ ia64_mca_cpe_move_page(u64 paddr, u32 node)
 
 	list_add(&page->lru, &pagelist);
 	SetPageMemError(page);		/* Mark the page as bad */
-	ret = migrate_pages(&pagelist, alloc_migrate_page, node);
+	ret = migrate_pages(&pagelist, alloc_migrate_page, node, false, true);
 	if (ret == 0) {
 		total_badpages++;
 		if (record_bad_paddr)
