From: Mel Gorman <mgorman@suse.de>
Date: Thu, 28 Apr 2011 09:40:50 +0100
Subject: [PATCH] mm: migrate: Migration accounts for its own isolated pages
References: Reclaim/compaction (fate#311931)
Patch-mainline: no, not required.

Signed-off-by: Mel Gorman <mgorman@suse.de>
---
 mm/migrate.c |    6 ------
 1 files changed, 0 insertions(+), 6 deletions(-)

diff --git a/mm/migrate.c b/mm/migrate.c
index b1a8e1b..8716378 100644
--- a/mm/migrate.c
+++ b/mm/migrate.c
@@ -788,12 +788,6 @@ int migrate_pages(struct list_head *from,
 	int rc;
 	unsigned long flags;
 
-	local_irq_save(flags);
-	list_for_each_entry(page, from, lru)
-		__inc_zone_page_state(page, NR_ISOLATED_ANON +
-				page_is_file_cache(page));
-	local_irq_restore(flags);
-
 	if (!swapwrite)
 		current->flags |= PF_SWAPWRITE;
 
