From: Fengguang Wu <fengguang.wu@intel.com>
Date: Tue, 18 Feb 2014 12:24:55 -0800
Patch-mainline: v3.15-rc1
Subject: Drivers: hv: fcopy_open() can be static
Git-commit: c765a6dfad10b857e9681d7c4dec375ebe78db8a
References: fate#317533

CC: "K. Y. Srinivasan" <kys@microsoft.com>
CC: "Greg Kroah-Hartman" <gregkh@linuxfoundation.org>
Signed-off-by: Fengguang Wu <fengguang.wu@intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Acked-by: <ohering@suse.de>
---
 drivers/hv/hv_fcopy.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/hv/hv_fcopy.c b/drivers/hv/hv_fcopy.c
index 285b5e2..eaaa3d8 100644
--- a/drivers/hv/hv_fcopy.c
+++ b/drivers/hv/hv_fcopy.c
@@ -327,7 +327,7 @@ static ssize_t fcopy_write(struct file *file, const char __user *buf,
 	return sizeof(int);
 }
 
-int fcopy_open(struct inode *inode, struct file *f)
+static int fcopy_open(struct inode *inode, struct file *f)
 {
 	/*
 	 * The user level daemon that will open this device is
@@ -344,7 +344,7 @@ int fcopy_open(struct inode *inode, struct file *f)
 	return 0;
 }
 
-int fcopy_release(struct inode *inode, struct file *f)
+static int fcopy_release(struct inode *inode, struct file *f)
 {
 	/*
 	 * The daemon has exited; reset the state.
