From: Fengguang Wu <fengguang.wu@intel.com>
Date: Fri, 18 Jan 2013 09:09:32 +0800
Patch-mainline: v3.9-rc1
Subject: Drivers: hv: vmbus_flow_handler() can be static
Git-commit: 83ebf6e562afc31cf4c6346ee87a7dd9fe39c322

Signed-off-by: Fengguang Wu <fengguang.wu@intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Acked-by: <ohering@suse.de>
---
 drivers/hv/vmbus_drv.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/hv/vmbus_drv.c b/drivers/hv/vmbus_drv.c
index 797142c..cf19dfa 100644
--- a/drivers/hv/vmbus_drv.c
+++ b/drivers/hv/vmbus_drv.c
@@ -509,7 +509,7 @@ static irqreturn_t vmbus_isr(int irq, void *dev_id)
  * can be handled concurrently.
  */
 
-void vmbus_flow_handler(unsigned int irq, struct irq_desc *desc)
+static void vmbus_flow_handler(unsigned int irq, struct irq_desc *desc)
 {
 	kstat_incr_irqs_this_cpu(irq, desc);
 
