From: Jiri Kosina <jkosina@suse.cz>
Subject: powerpc/barrier: add gmb
Patch-mainline: Not yet, under development
References: bsc#1068032 CVE-2017-5753 

Signed-off-by: Jiri Kosina <jkosina@suse.cz>

--- a/arch/powerpc/include/asm/system.h
+++ b/arch/powerpc/include/asm/system.h
@@ -42,6 +42,8 @@
 
 #define set_mb(var, value)	do { var = value; mb(); } while (0)
 
+#define gmb() asm volatile("ori 31,31,0")
+
 #ifdef __KERNEL__
 #define AT_VECTOR_SIZE_ARCH 6 /* entries in ARCH_DLINFO */
 #ifdef CONFIG_SMP
