From: Andreas Gerstmayr <andreas.gerstmayr@gmail.com>
Date: Thu, 15 Sep 2016 21:23:01 +0200
Subject: ceph: fix description for rsize and rasize mount options
Git-commit: 92c1037ce8e3a70b616c8b91c990caf16a0096d1
Patch-mainline: v4.9-rc1
References: FATE#322288

Signed-off-by: Andreas Gerstmayr <andreas.gerstmayr@gmail.com>
Signed-off-by: Ilya Dryomov <idryomov@gmail.com>
Acked-by: Luis Henriques <lhenriques@suse.com>
---
 Documentation/filesystems/ceph.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Documentation/filesystems/ceph.txt b/Documentation/filesystems/ceph.txt
index d6030aa33376..f5306ee40ea9 100644
--- a/Documentation/filesystems/ceph.txt
+++ b/Documentation/filesystems/ceph.txt
@@ -98,6 +98,10 @@ Mount Options
 	size.
 
   rsize=X
+	Specify the maximum read size in bytes.  By default there is no
+	maximum.
+
+  rasize=X
 	Specify the maximum readahead.
 
   mount_timeout=X

