From: "Yan, Zheng" <zyan@redhat.com>
Date: Fri, 29 Apr 2016 23:40:23 +0800
Subject: ceph: fix inode reference leak
Git-commit: 209ae762a64397557be2bc48490fd968912e9a71
Patch-mainline: v4.7-rc1
References: FATE#322288

Signed-off-by: Yan, Zheng <zyan@redhat.com>
Acked-by: Luis Henriques <lhenriques@suse.com>
---
 fs/ceph/inode.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/ceph/inode.c b/fs/ceph/inode.c
index f51b6fd5f570..4889aaa72acc 100644
--- a/fs/ceph/inode.c
+++ b/fs/ceph/inode.c
@@ -1328,8 +1328,8 @@ static int readdir_prepopulate_inodes_only(struct ceph_mds_request *req,
 		if (rc < 0) {
 			pr_err("fill_inode badness on %p got %d\n", in, rc);
 			err = rc;
-			continue;
 		}
+		iput(in);
 	}
 
 	return err;

