From: Ilya Dryomov <idryomov@gmail.com>
Date: Fri, 23 Sep 2016 09:57:56 +0200
Subject: libceph: use CEPH_AUTH_UNKNOWN in ceph_auth_build_hello()
Git-commit: fdc723e77bcf17ec2646fdef031e7f135968a9b7
Patch-mainline: v4.9-rc1
References: FATE#322288

Signed-off-by: Ilya Dryomov <idryomov@gmail.com>
Acked-by: Luis Henriques <lhenriques@suse.com>
---
 net/ceph/auth.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/ceph/auth.c b/net/ceph/auth.c
index 78067dda9d3c..53aa2d27e36b 100644
--- a/net/ceph/auth.c
+++ b/net/ceph/auth.c
@@ -114,7 +114,7 @@ int ceph_auth_build_hello(struct ceph_auth_client *ac, void *buf, size_t len)
 	monhdr->session_mon = cpu_to_le16(-1);
 	monhdr->session_mon_tid = 0;
 
-	ceph_encode_32(&p, 0);  /* no protocol, yet */
+	ceph_encode_32(&p, CEPH_AUTH_UNKNOWN);  /* no protocol, yet */
 
 	lenp = p;
 	p += sizeof(u32);

