From: <ohering@suse.de>
Subject: hv: only load the driver in a hyper-v guest
References: bnc#704957

---
 drivers/staging/hv/vmbus_drv.c |   19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

--- a/drivers/staging/hv/vmbus_drv.c
+++ b/drivers/staging/hv/vmbus_drv.c
@@ -30,6 +30,7 @@
 #include <linux/sysctl.h>
 #include <linux/slab.h>
 #include <linux/acpi.h>
+#include <linux/dmi.h>
 #include <acpi/acpi_bus.h>
 #include <linux/completion.h>
 
@@ -722,10 +723,28 @@ static struct acpi_driver vmbus_acpi_dri
 	},
 };
 
+static const struct dmi_system_id __initconst
+hv_vmbus_dmi_table[] __maybe_unused  = {
+	{
+		.ident = "Hyper-V",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "Microsoft Corporation"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "Virtual Machine"),
+			DMI_MATCH(DMI_BOARD_NAME, "Virtual Machine"),
+		},
+	},
+	{ },
+};
+MODULE_DEVICE_TABLE(dmi, hv_vmbus_dmi_table);
+
 static int __init hv_acpi_init(void)
 {
 	int ret, t;
 
+	/* Do not stall 5 seconds */
+	if (!dmi_check_system(hv_vmbus_dmi_table))
+		return -ENODEV;
+
 	init_completion(&probe_event);
 
 	/*
