From: <ohering@suse.de>
Subject: hv: only load the driver in a hyper-v guest
References: bnc#704957

---
 drivers/staging/hv/vmbus_drv.c |   17 +++++++++++++++++
 1 file changed, 17 insertions(+)

--- a/drivers/staging/hv/vmbus_drv.c
+++ b/drivers/staging/hv/vmbus_drv.c
@@ -766,10 +766,27 @@ static const struct pci_device_id micros
 };
 MODULE_DEVICE_TABLE(pci, microsoft_hv_pci_table);
 
+static const struct dmi_system_id __initconst
+hv_vmbus_dmi_table[] __maybe_unused  = {
+	{
+		.ident = "Hyper-V",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "Microsoft Corporation"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "Virtual Machine"),
+			DMI_MATCH(DMI_BOARD_NAME, "Virtual Machine"),
+		},
+	},
+	{ },
+};
+MODULE_DEVICE_TABLE(dmi, hv_vmbus_dmi_table);
+
 static int __init hv_acpi_init(void)
 {
 	int ret;
 
+	if (!dmi_check_system(hv_vmbus_dmi_table))
+		return -ENODEV;
+
 	init_completion(&probe_event);
 
 	/*
