From: <ohering@suse.de>
Subject: hv: Actually show the modalias files in /sys/bus/vmbus/devices/*/
References: fate#311487

---
 drivers/staging/hv/vmbus_drv.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/staging/hv/vmbus_drv.c
+++ b/drivers/staging/hv/vmbus_drv.c
@@ -127,6 +127,8 @@ static ssize_t vmbus_show_device_attr(st
 
 	if (!strcmp(dev_attr->attr.name, "class_id")) {
 		return sprintf(buf, "%s\n", hv_dev->device_type);
+	} else if (!strcmp(dev_attr->attr.name, "modalias")) {
+		return sprintf(buf, "vmbus:%s\n", hv_dev->device_type);
 	} else if (!strcmp(dev_attr->attr.name, "state")) {
 		return sprintf(buf, "%d\n", device_info.chn_state);
 	} else if (!strcmp(dev_attr->attr.name, "id")) {
@@ -181,6 +183,7 @@ static struct device_attribute vmbus_dev
 	__ATTR(id, S_IRUGO, vmbus_show_device_attr, NULL),
 	__ATTR(state, S_IRUGO, vmbus_show_device_attr, NULL),
 	__ATTR(class_id, S_IRUGO, vmbus_show_device_attr, NULL),
+	__ATTR(modalias, S_IRUGO, vmbus_show_device_attr, NULL),
 	__ATTR(monitor_id, S_IRUGO, vmbus_show_device_attr, NULL),
 
 	__ATTR(server_monitor_pending, S_IRUGO, vmbus_show_device_attr, NULL),
