Subject: staging-hv: bind old hv_mouse to hid-hyperv
From: <ohering@suse.de>
Patch-mainline: never

---
 drivers/hid/hid-hyperv.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/hid/hid-hyperv.c
+++ b/drivers/hid/hid-hyperv.c
@@ -583,5 +583,7 @@ static void __exit mousevsc_exit(void)
 }
 
 MODULE_LICENSE("GPL");
+/* bind to old driver to simplify upgrade path */
+MODULE_ALIAS("hv_mouse");
 module_init(mousevsc_init);
 module_exit(mousevsc_exit);
