Subject: hv: print extra debug in kvp_on_msg in error paths
From: <ohering@suse.de>
Patch-mainline: Never, this change is just for diagnostic
References: bnc#1039153

--- a/drivers/hv/hv_kvp.c
+++ b/drivers/hv/hv_kvp.c
@@ -202,7 +202,14 @@ static int kvp_on_msg(void *msg, int len
 	int	error = 0;
 
 	if (len < sizeof(*message))
+	{
+		static int i = 0;
+		if (!i) {
+			i = 1;
+			pr_err("bug#1039153 %s(%u) %s[%u], len %d msg %d state %d\n", __func__, __LINE__, current->comm, current->pid, len, (int)sizeof(*message), kvp_transaction.state);
+		}
 		return -EINVAL;
+	}
 
 	/*
 	 * If we are negotiating the version information
@@ -215,7 +222,14 @@ static int kvp_on_msg(void *msg, int len
 
 	/* We didn't send anything to userspace so the reply is spurious */
 	if (kvp_transaction.state < HVUTIL_USERSPACE_REQ)
+	{
+		static int i = 0;
+		if (!i) {
+			i = 1;
+			pr_err("bug#1039153 %s(%u) %s[%u], state %d < %d\n", __func__, __LINE__, current->comm, current->pid, kvp_transaction.state, HVUTIL_USERSPACE_REQ);
+		}
 		return -EINVAL;
+	}
 
 	kvp_transaction.state = HVUTIL_USERSPACE_RECV;
 
