From: <ohering@suse.de>
Date: Fri, 11 May 2012 22:16:08 +0000
Patch-mainline: v3.5-rc1
Subject: [PATCH] hv: add RNDIS_OID_GEN_RNDIS_CONFIG_PARAMETER
Git-commit: 8cdddc3f9d6692f5690841468a9f63c19fb734ed

---
 drivers/net/hyperv/hyperv_net.h |    3 +++
 1 file changed, 3 insertions(+)

Index: linux-3.1-openSUSE-12.1/drivers/net/hyperv/hyperv_net.h
===================================================================
--- linux-3.1-openSUSE-12.1.orig/drivers/net/hyperv/hyperv_net.h
+++ linux-3.1-openSUSE-12.1/drivers/net/hyperv/hyperv_net.h
@@ -25,6 +25,9 @@
 #ifndef _HYPERV_NET_H
 #define _HYPERV_NET_H
 
+/* from include/linux/rndis.h via 8cdddc3f9d6692f5690841468a9f63c19fb734ed */
+#define RNDIS_OID_GEN_RNDIS_CONFIG_PARAMETER   0x0001021B
+
 #include <linux/list.h>
 #include <linux/hyperv.h>
 #include <linux/rndis.h>
