From d58b1cacce8a7ad27139179d8a476da9bba9e4b0 Mon Sep 17 00:00:00 2001
From: Ben Hutchings <bhutchings@solarflare.com>
Date: Tue, 10 Jul 2012 10:55:35 +0000
Subject: [PATCH 145/183] net: Fix non-kernel-doc comments with kernel-doc
 start marker
Git-commit: d58b1cacce8a7ad27139179d8a476da9bba9e4b0
Patch-mainline: Merged into tipc-devel
References: bnc#797455

Signed-off-by: Ben Hutchings <bhutchings@solarflare.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Erik Hugne <erik.hugne@ericsson.com>
Signed-off-by: Jiri Kosina <jkosina@suse.cz>

---
 net/tipc/socket.c |    5 ++---
 1 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/net/tipc/socket.c b/net/tipc/socket.c
index 5f8fbd7..572d096 100644
--- a/net/tipc/socket.c
+++ b/net/tipc/socket.c
@@ -1698,9 +1698,8 @@ static int getsockopt(struct socket *sock,
 	return put_user(sizeof(value), ol);
 }
 
-/**
- * Protocol switches for the various types of TIPC sockets
- */
+/* Protocol switches for the various types of TIPC sockets */
+
 static const struct proto_ops msg_ops = {
 	.owner		= THIS_MODULE,
 	.family		= AF_TIPC,
-- 
1.7.8.3

