From: Russ Anderson <rja@sgi.com>
Subject: x86 / UV: Add KDB support to UV NMI handler
References: bnc#579647, FATE#306952
Patch-mainline: no

Signed-off-by: Rafael J. Wysocki <rjw@suse.de>
---
 arch/x86/kernel/apic/x2apic_uv_x.c |   23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

--- a/arch/x86/kernel/apic/x2apic_uv_x.c
+++ b/arch/x86/kernel/apic/x2apic_uv_x.c
@@ -21,6 +21,9 @@
 #include <linux/init.h>
 #include <linux/io.h>
 #include <linux/kdebug.h>
+#ifdef CONFIG_KDB
+#include <linux/kdb.h>
+#endif
 
 #include <asm/uv/uv_mmrs.h>
 #include <asm/uv/uv_hub.h>
@@ -579,8 +582,27 @@ void __cpuinit uv_cpu_init(void)
  */
 int uv_handle_nmi(struct notifier_block *self, unsigned long reason, void *data)
 {
+#ifdef CONFIG_KDB
+	struct die_args *args = data;
+	struct pt_regs *regs = args->regs;
+	static int controlling_cpu = -1;
+#endif
+
 	if (reason != DIE_NMI_IPI)
 		return NOTIFY_OK;
+
+#ifdef CONFIG_KDB
+	spin_lock(&uv_nmi_lock);
+	if (controlling_cpu == -1) {
+		controlling_cpu = smp_processor_id();
+		spin_unlock(&uv_nmi_lock);
+		(void)kdb(KDB_REASON_NMI, reason, regs);
+		controlling_cpu = -1;
+	} else {
+		spin_unlock(&uv_nmi_lock);
+		(void)kdb(KDB_REASON_ENTER_SLAVE, reason, regs);
+	}
+#else
 	/*
 	 * Use a lock so only one cpu prints at a time
 	 * to prevent intermixed output.
@@ -589,6 +611,7 @@ int uv_handle_nmi(struct notifier_block
 	pr_info("NMI stack dump cpu %u:\n", smp_processor_id());
 	dump_stack();
 	spin_unlock(&uv_nmi_lock);
+#endif /* CONFIG_KDB */
 
 	return NOTIFY_STOP;
 }
