From: Hannes Reinecke <hare@suse.de>
Subject: Xen host lock push-down
References: FATE#311826
Patch-Mainline: n/a

Update Xen drivers for the host lock push-down.

Signed-off-by: Hannes Reinecke <hare@suse.de>

diff --git a/drivers/xen/scsifront/scsifront.c b/drivers/xen/scsifront/scsifront.c
index 2db0ccd..fd4e60e 100644
--- a/drivers/xen/scsifront/scsifront.c
+++ b/drivers/xen/scsifront/scsifront.c
@@ -315,8 +315,10 @@ big_to_sg:
 	return ref_cnt;
 }
 
-static int scsifront_queuecommand(struct scsi_cmnd *sc,
-				  void (*done)(struct scsi_cmnd *))
+static int scsifront_queuecommand(struct Scsi_Host *, struct scsi_cmnd *);
+
+static int scsifront_queuecommand_lck(struct scsi_cmnd *sc,
+				      void (*done)(struct scsi_cmnd *))
 {
 	struct vscsifrnt_info *info = shost_priv(sc->device->host);
 	vscsiif_request_t *ring_req;
@@ -384,6 +386,7 @@ out_fail_command:
 	return 0;
 }
 
+static DEF_SCSI_QCMD(scsifront_queuecommand);
 
 static int scsifront_eh_abort_handler(struct scsi_cmnd *sc)
 {
