From: Joerg Roedel <jroedel@suse.de>
Date: Wed, 9 May 2018 21:38:50 +0200
Subject: SSB kABI Fix
Patch-mainline: Not yet, work in progress
References: bsc#1087082 CVE-2018-3639

Signed-off-by: Joerg Roedel <jroedel@suse.de>
---
 include/linux/sched.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/linux/sched.h b/include/linux/sched.h
index f8cde5ee..5d8cbd25 100644
--- a/include/linux/sched.h
+++ b/include/linux/sched.h
@@ -1383,8 +1383,10 @@ struct task_struct {
 	unsigned sched_reset_on_fork:1;
 	unsigned sched_contributes_to_load:1;
 
+#ifndef __GENKSYMS__
 	unsigned ssb_disable:1;
 	unsigned ssb_force_disable:1;
+#endif
 
 	pid_t pid;
 	pid_t tgid;
-- 
2.12.3

