From: NeilBrown <neilb@suse.com>
Subject: KABI fix
Patch-mainline: Never, kabi
References: bsc#1014410

Acked-by: NeilBrown <neilb@suse.com>
---
 include/linux/sunrpc/xprt.h |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

--- a/include/linux/sunrpc/xprt.h
+++ b/include/linux/sunrpc/xprt.h
@@ -211,8 +211,10 @@ struct rpc_xprt {
 	struct work_struct	task_cleanup;
 	struct timer_list	timer;
 	unsigned long		last_used,
-				idle_timeout,
-				max_reconnect_timeout;
+				idle_timeout;
+#ifndef __GENKSYMS__
+	unsigned long		max_reconnect_timeout;
+#endif
 
 	/*
 	 * Send stuff
