From: jbeulich@suse.com
Subject: netfront: fix kABI after "reduce gso_max_size to account for max TCP header"
Patch-mainline: Never (kABI fix)

---
 drivers/xen/netfront/netfront.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/xen/netfront/netfront.c
+++ b/drivers/xen/netfront/netfront.c
@@ -51,7 +51,9 @@
 #include <net/sock.h>
 #include <net/pkt_sched.h>
 #include <net/route.h>
+#ifndef __GENKSYMS__
 #include <net/tcp.h>
+#endif
 #include <asm/uaccess.h>
 #include <xen/evtchn.h>
 #include <xen/xenbus.h>
