From: Miklos Szeredi <mszeredi@suse.cz>
Subject: epoll: hide kabi change in struct file
Patch-mainline: never
References: bnc#676204 CVE-2011-1083

Signed-off-by: Miklos Szeredi <mszeredi@suse.cz>
---
 include/linux/fs.h |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

Index: linux-3.0-SLE11-SP2/include/linux/fs.h
===================================================================
--- linux-3.0-SLE11-SP2.orig/include/linux/fs.h	2012-03-05 23:10:57.000000000 +0100
+++ linux-3.0-SLE11-SP2/include/linux/fs.h	2012-03-05 23:14:04.000000000 +0100
@@ -978,12 +978,14 @@ struct file {
 #ifdef CONFIG_EPOLL
 	/* Used by fs/eventpoll.c to link all the hooks to this file */
 	struct list_head	f_ep_links;
-	struct list_head	f_tfile_llink;
 #endif /* #ifdef CONFIG_EPOLL */
 	struct address_space	*f_mapping;
 #ifdef CONFIG_DEBUG_WRITECOUNT
 	unsigned long f_mnt_write_state;
 #endif
+#if defined(CONFIG_EPOLL) && !defined(__GENKSYMS__)
+	struct list_head	f_tfile_llink;
+#endif
 };
 
 struct file_handle {
