From: Tony Jones <tonyj@suse.de>
Subject: Ignore 'flags' change to event_constraint
Date: Tue May  6 18:00:06 PDT 2014
References: bnc#876114, fate#317174
Patch-mainline: Never, kabi
Signed-off-by: Tony Jones <tonyj@suse.de>

---
 arch/x86/kernel/cpu/perf_event.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/arch/x86/kernel/cpu/perf_event.h
+++ b/arch/x86/kernel/cpu/perf_event.h
@@ -60,7 +60,9 @@ struct event_constraint {
 	u64	cmask;
 	int	weight;
 	int	overlap;
+#ifndef __GENKSYMS__
 	int	flags;
+#endif
 };
 /*
  * struct event_constraint flags
