From: Jiri Slaby <jslaby@suse.cz>
Subject: Revert nr_forced2_migrations removal from sched_entity
Patch-mainline: never

It breaks kabi.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/linux/sched.h |    1 +
 1 file changed, 1 insertion(+)

--- a/include/linux/sched.h
+++ b/include/linux/sched.h
@@ -1182,6 +1182,7 @@ struct sched_entity {
 	u64			nr_failed_migrations_running;
 	u64			nr_failed_migrations_hot;
 	u64			nr_forced_migrations;
+	u64			nr_forced2_migrations;
 
 	u64			nr_wakeups;
 	u64			nr_wakeups_sync;
