Subject: sched: guard struct sched_group kABI changes
From: Jiri Slaby <jslaby@suse.cz>
Patch-mainline: never

silence kABI checker.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
---
 include/linux/sched.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/include/linux/sched.h
+++ b/include/linux/sched.h
@@ -894,7 +894,9 @@ struct sched_group {
 	 * single CPU.
 	 */
 	unsigned int cpu_power;
+#ifndef __GENKSYMS__
 	unsigned int group_weight;
+#endif
 
 	/*
 	 * The CPUs this group covers.
