From: Jiri Slaby <jslaby@suse.cz>
Subject: Fix task_struct kabi breakage
Patch-mainline: never
References: bnc#598677

Introduced by
 patch-2.6.32.14

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 include/linux/sched.h |    5 +++++
 1 file changed, 5 insertions(+)

--- a/include/linux/sched.h
+++ b/include/linux/sched.h
@@ -1545,6 +1545,11 @@ struct task_struct {
 	/* bitmask of trace recursion */
 	unsigned long trace_recursion;
 #endif /* CONFIG_TRACING */
+#ifdef __GENKSYMS__
+	unsigned long stack_start; /* nobody should use this */
+#else
+	unsigned long ___________unused;
+#endif
 #ifdef CONFIG_CGROUP_MEM_RES_CTLR /* memcg uses this to do batch job */
 	struct memcg_batch_info {
 		int do_batch;	/* incremented when batch uncharge started */
