From: Jan Kara <jack@suse.cz>
Subject: [PATCH] Fixup kABI after patches.fixes/writeback-do-not-sync-data-dirtied-after-sync-start.patch
Patch-mainline: Never
References: bnc#833820

Signed-off-by: Jan Kara <jack@suse.cz>

Index: linux-3.0-SLE11-SP3/include/linux/backing-dev.h
===================================================================
--- linux-3.0-SLE11-SP3.orig/include/linux/backing-dev.h
+++ linux-3.0-SLE11-SP3/include/linux/backing-dev.h
@@ -48,7 +48,11 @@ enum bdi_stat_item {
 
 struct bdi_writeback {
 	struct backing_dev_info *bdi;	/* our parent bdi */
+#ifndef __GENKSYMS__
 	unsigned long last_move_tail;
+#else
+	unsigned int nr;
+#endif
 	unsigned long last_old_flush;	/* last old data flush */
 	unsigned long last_active;	/* last time bdi thread was active */
 
