From: jbeulich@novell.com
Subject: suppress kABI change resulting from effectively renaming irq_cpustat_t.irq_tlb_count (unused on Xen) to irq_lock_count
Patch-mainline: n/a

--- sle11sp1-2010-08-06.orig/arch/x86/include/asm/hardirq.h	2010-08-17 10:38:13.000000000 +0200
+++ sle11sp1-2010-08-06/arch/x86/include/asm/hardirq.h	2010-08-17 16:03:03.000000000 +0200
@@ -21,6 +21,9 @@ typedef struct {
 #ifndef CONFIG_XEN
 	unsigned int irq_tlb_count;
 #else
+#ifdef __GENKSYMS__
+#define irq_lock_count irq_tlb_count
+#endif
 	unsigned int irq_lock_count;
 #endif
 #endif
