From: Joerg Roedel <jroedel@suse.de>
Date: Wed, 9 May 2018 22:05:33 +0200
Subject: SSB: Xen build fix
Patch-mainline: Not yet, work in progress
References: bsc#1087082 CVE-2018-3639

Signed-off-by: Joerg Roedel <jroedel@suse.de>
---
 arch/x86/kernel/process-xen.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/x86/kernel/process-xen.c b/arch/x86/kernel/process-xen.c
index 3a8a891a..589121c9 100644
--- a/arch/x86/kernel/process-xen.c
+++ b/arch/x86/kernel/process-xen.c
@@ -193,6 +193,11 @@ int set_tsc_mode(unsigned int val)
 	return 0;
 }
 
+void speculative_store_bypass_update(void)
+{
+	/* Nothing to do for Xen here */
+}
+
 void __switch_to_xtra(struct task_struct *prev_p, struct task_struct *next_p)
 {
 	struct thread_struct *prev, *next;
-- 
2.12.3

