From http://xenbits.xen.org/hg/linux-2.6.18-xen.hg/rev/b9627a5740d2
From: xen-devel@lists.xen.org
Patch-mainline: n/a
Subject: xenoprof: backward compatibility for changed XENOPROF_ESCAPE_CODE

Signed-off-by: Jan Beulich <jbeulich@suse.com>

(Also include the definition change itself.)

--- sle11sp2.orig/drivers/xen/xenoprof/xenoprofile.c	2012-02-27 12:58:13.000000000 +0100
+++ sle11sp2/drivers/xen/xenoprof/xenoprofile.c	2012-02-27 12:58:19.000000000 +0100
@@ -120,6 +120,10 @@ static struct xenoprof_passive passive_d
 /* Check whether the given entry is an escape code */
 static int xenoprof_is_escape(xenoprof_buf_t * buf, int tail)
 {
+#if CONFIG_XEN_COMPAT < 0x040200 && !defined(CONFIG_64BIT)
+	if (buf->event_log[tail].eip == (unsigned long)XENOPROF_ESCAPE_CODE)
+		return 1;
+#endif
 	return (buf->event_log[tail].eip == XENOPROF_ESCAPE_CODE);
 }
 
--- sle11sp2.orig/include/xen/interface/xenoprof.h	2011-09-06 08:32:24.000000000 +0200
+++ sle11sp2/include/xen/interface/xenoprof.h	2012-02-27 12:58:19.000000000 +0100
@@ -68,7 +68,7 @@ struct event_log {
 };
 
 /* PC value that indicates a special code */
-#define XENOPROF_ESCAPE_CODE ~0UL
+#define XENOPROF_ESCAPE_CODE (~0ULL)
 /* Transient events for the xenoprof->oprofile cpu buf */
 #define XENOPROF_TRACE_BEGIN 1
 
