From http://xenbits.xen.org/hg/linux-2.6.18-xen.hg/rev/341104e962db
From: xen-devel@lists.xen.org
Patch-mainline: n/a
Subject: x86-64: fix hypercall page unwind info

This apparently was a copy-and-paste oversight.

Signed-off-by: Jan Beulich <jbeulich@suse.com>

--- sle11sp2.orig/arch/x86/kernel/head_64-xen.S	2012-09-04 16:31:55.000000000 +0200
+++ sle11sp2/arch/x86/kernel/head_64-xen.S	2012-09-03 15:31:18.000000000 +0200
@@ -86,7 +86,7 @@ NEXT_PAGE(hypercall_page)
 	CFI_REL_OFFSET	rcx,0
 	.skip 2 /* push %r11 */
 	CFI_ADJUST_CFA_OFFSET	8
-	CFI_REL_OFFSET	rcx,0
+	CFI_REL_OFFSET	r11,0
 	.skip 5 /* mov $#,%eax */
 	.skip 2 /* syscall */
 	.skip 2 /* pop %r11 */
