From http://xenbits.xen.org/hg/linux-2.6.18-xen.hg/rev/9db2ee23611e
From: xen-devel@lists.xen.org
Patch-mainline: n/a
Subject: xenbus: fix overflow check in xenbus_dev_write()

Reported-by: Dan Carpenter <dan.carpenter@oracle.com>
Signed-off-by: Jan Beulich <jbeulich@suse.com>
Acked-by: Ian Campbell <ian.campbell@citrix.com>

--- sle11sp2.orig/drivers/xen/xenbus/xenbus_dev.c	2012-02-07 15:25:58.000000000 +0100
+++ sle11sp2/drivers/xen/xenbus/xenbus_dev.c	2012-10-25 10:27:40.000000000 +0200
@@ -239,7 +239,7 @@ static ssize_t xenbus_dev_write(struct f
 	if (!is_xenstored_ready())
 		return -ENODEV;
 
-	if ((len + u->len) > sizeof(u->u.buffer)) {
+	if (len > sizeof(u->u.buffer) - u->len) {
 		rc = -EINVAL;
 		goto out;
 	}
