From http://xenbits.xen.org/hg/linux-2.6.18-xen.hg/rev/f8a110b3f989
From: xen-devel@lists.xenproject.org
Patch-mainline: Never, SUSE-Xen specific
Subject: balloon: don't BUG() when balloon is empty
References: bsc#1083347

Signed-off-by: Jan Beulich <jbeulich@suse.com>

--- a/drivers/xen/balloon/balloon.c
+++ b/drivers/xen/balloon/balloon.c
@@ -254,13 +254,18 @@ static int increase_reservation(unsigned
 
 	page = balloon_first_page();
 	for (i = 0; i < nr_pages; i++) {
-		BUG_ON(page == NULL);
-		frame_list[i] = page_to_pfn(page);;
+		if (!page) {
+			if (i)
+				break;
+			rc = -ENXIO;
+			goto out;
+		}
+		frame_list[i] = page_to_pfn(page);
 		page = balloon_next_page(page);
 	}
 
 	set_xen_guest_handle(reservation.extent_start, frame_list);
-	reservation.nr_extents = nr_pages;
+	reservation.nr_extents = i;
 	rc = HYPERVISOR_memory_op(XENMEM_populate_physmap, &reservation);
 	if (rc < 0)
 		goto out;
