Git-commit: 07c540a0b5f4674538b57ad85bc9306e44fb45dd
From: Roger Pau Monne <roger.pau@citrix.com>
Subject: xen-blkfront: free allocated page
Patch-mainline: 3.8-rc7
References: bsc#961658 fate#320200

Free the page allocated for the persistent grant.

Signed-off-by: Roger Pau Monné <roger.pau@citrix.com>
Signed-off-by: Konrad Rzeszutek Wilk <konrad.wilk@oracle.com>
Acked-by: jbeulich@suse.com

--- a/drivers/block/xen-blkfront.c
+++ b/drivers/block/xen-blkfront.c
@@ -781,6 +781,7 @@ static void blkif_free(struct blkfront_i
 		all_gnts = llist_del_all(&info->persistent_gnts);
 		llist_for_each_entry(persistent_gnt, all_gnts, node) {
 			gnttab_end_foreign_access(persistent_gnt->gref, 0, 0UL);
+			__free_page(pfn_to_page(persistent_gnt->pfn));
 			kfree(persistent_gnt);
 		}
 		info->persistent_gnts_c = 0;
