From: Hannes Reinecke <hare@suse.de>
Subject: Change reserve_ibft_region() to find_ibft_region()
References: FATE#311488
Patch-Mainline: n/a

This is a port of commit 042be38e6106ed70b42d096ab4a1ed4187e510e6
to apply for Xen, too.

Signed-off-by: Hannes Reinecke <hare@suse.de>

diff --git a/arch/x86/kernel/setup-xen.c b/arch/x86/kernel/setup-xen.c
index 838e45e..67e4f24 100644
--- a/arch/x86/kernel/setup-xen.c
+++ b/arch/x86/kernel/setup-xen.c
@@ -693,6 +693,16 @@ static int __init setup_elfcorehdr(char *arg)
 early_param("elfcorehdr", setup_elfcorehdr);
 #endif
 
+static __init void reserve_ibft_region(void)
+{
+	unsigned long addr, size = 0;
+
+	addr = find_ibft_region(&size);
+
+	if (size)
+		reserve_early_overlap_ok(addr, addr + size, "ibft");
+}
+
 #ifdef CONFIG_X86_RESERVE_LOW_64K
 static int __init dmi_low_memory_corruption(const struct dmi_system_id *d)
 {
