From 2998b69687ad68bc39c9357fe7ab7f247274f6d7 Mon Sep 17 00:00:00 2001
From: Joerg Roedel <jroedel@suse.de>
Date: Tue, 13 Mar 2018 15:10:29 +0100
Subject: xen/x86/mm: Move KAISER functions from pgtable_64.h to pgtable.h
References: bsc#1068032 CVE-2017-5754
Patch-mainline: Never, SUSE-Xen specific

Make them usable from 32 and 64 bit code.

Signed-off-by: Joerg Roedel <jroedel@suse.de>

--- a/arch/x86/include/mach-xen/asm/pgtable.h
+++ b/arch/x86/include/mach-xen/asm/pgtable.h
@@ -388,6 +388,29 @@ static inline int is_new_memtype_allowed
 
 pmd_t *populate_extra_pmd(unsigned long vaddr);
 pte_t *populate_extra_pte(unsigned long vaddr);
+
+#ifdef CONFIG_KAISER
+extern pgd_t kaiser_set_shadow_pgd(pgd_t *pgdp, pgd_t pgd);
+
+static inline pgd_t *xen_get_shadow_pgd(pgd_t *pgdp)
+{
+#ifdef CONFIG_DEBUG_VM
+	/* linux/mmdebug.h may not have been included at this point */
+	BUG_ON(!kaiser_enabled);
+#endif
+	return (pgd_t *)((unsigned long)pgdp | (unsigned long)PAGE_SIZE);
+}
+#else
+static inline pgd_t kaiser_set_shadow_pgd(pgd_t *pgdp, pgd_t pgd)
+{
+	return pgd;
+}
+static inline pgd_t *xen_get_shadow_pgd(pgd_t *pgdp)
+{
+	return NULL;
+}
+#endif /* CONFIG_KAISER */
+
 #endif	/* __ASSEMBLY__ */
 
 #ifdef CONFIG_X86_32
@@ -555,6 +578,8 @@ static inline int pud_large(pud_t pud)
 }
 #endif	/* PAGETABLE_LEVELS > 2 */
 
+static inline int pgd_large(pgd_t pgd) { return 0; }
+
 #if PAGETABLE_LEVELS > 3
 static inline int pgd_present(pgd_t pgd)
 {
--- a/arch/x86/include/mach-xen/asm/pgtable_64.h
+++ b/arch/x86/include/mach-xen/asm/pgtable_64.h
@@ -102,28 +102,6 @@ static inline void xen_pud_clear(pud_t *
 
 #define __user_pgd(pgd) ((pgd) + PTRS_PER_PGD)
 
-#ifdef CONFIG_KAISER
-extern pgd_t kaiser_set_shadow_pgd(pgd_t *pgdp, pgd_t pgd);
-
-static inline pgd_t *xen_get_shadow_pgd(pgd_t *pgdp)
-{
-#ifdef CONFIG_DEBUG_VM
-	/* linux/mmdebug.h may not have been included at this point */
-	BUG_ON(!kaiser_enabled);
-#endif
-	return (pgd_t *)((unsigned long)pgdp | (unsigned long)PAGE_SIZE);
-}
-#else
-static inline pgd_t kaiser_set_shadow_pgd(pgd_t *pgdp, pgd_t pgd)
-{
-	return pgd;
-}
-static inline pgd_t *xen_get_shadow_pgd(pgd_t *pgdp)
-{
-	return NULL;
-}
-#endif /* CONFIG_KAISER */
-
 static inline void xen_set_pgd(pgd_t *pgdp, pgd_t pgd)
 {
 	xen_l4_entry_update(pgdp, kaiser_set_shadow_pgd(pgdp, pgd));
@@ -147,7 +125,6 @@ extern void sync_global_pgds(unsigned lo
 /*
  * Level 4 access.
  */
-static inline int pgd_large(pgd_t pgd) { return 0; }
 #define mk_kernel_pgd(address) __pgd((address) | _KERNPG_TABLE)
 
 /* PUD - Level3 access */
