From: Kevin Winchester <kjwinchester@gmail.com>
Subject: x86, perf: Clean up perf_event cpu code
References: bnc#789200, fate#313753, fate#312066, fate#313762
Patch-mainline: v3.2-rc1
Signed-off-by: Tony Jones <tonyj@suse.de>

--- sle11sp3.orig/arch/x86/kernel/cpu/Makefile	2012-12-11 14:54:27.000000000 +0100
+++ sle11sp3/arch/x86/kernel/cpu/Makefile	2012-08-20 13:22:01.000000000 +0200
@@ -42,7 +42,7 @@ obj-$(CONFIG_MTRR)			+= mtrr/
 obj-$(CONFIG_X86_LOCAL_APIC)		+= perfctr-watchdog.o
 
 disabled-obj-$(CONFIG_XEN) := hypervisor.o mshyperv.o perfctr-watchdog.o \
-			      perf_event.o sched.o vmware.o
+			      perf_event.o perf_event_%.o sched.o vmware.o
 
 quiet_cmd_mkcapflags = MKCAP   $@
       cmd_mkcapflags = $(PERL) $(srctree)/$(src)/mkcapflags.pl $< $@
